<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />



<meta name="date" content="2019-03-13" />

<title>Preprocessing</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<div class="container-fluid main-container">

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">MAGL</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Preprocessing</h1>
<h4 class="author"><em>Pierre-Luc Germain</em></h4>
<address class="author_afil">
Institute for Molecular Life Sciences, University of Zurich, Switzerland<br><h4 class="author"><em>Helena L. Crowell</em></h4>
<address class="author_afil">
Institute for Molecular Life Sciences, University of Zurich, Switzerland<br><h4 class="date"><em>March 13, 2019</em></h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#report"> Report <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-03-13
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>MAGL/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.2.0). The <em>Report</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="report" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190311code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190311)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190311code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190311)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomHelenaLCmuscatcomparisontreeeaed7ecf04d1ff5ca101c68dfa7079d114b49658targetblankeaed7eca"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/HelenaLC/muscat-comparison/tree/eaed7ecf04d1ff5ca101c68dfa7079d114b49658" target="_blank">eaed7ec</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomHelenaLCmuscatcomparisontreeeaed7ecf04d1ff5ca101c68dfa7079d114b49658targetblankeaed7eca" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    MAGL/.DS_Store
    Ignored:    MAGL/analysis/.DS_Store
    Ignored:    MAGL/data/
    Ignored:    MAGL/output/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/HelenaLC/muscat-comparison/blob/eaed7ecf04d1ff5ca101c68dfa7079d114b49658/MAGL/analysis/0-preprocessing.Rmd" target="_blank">eaed7ec</a>
</td>
<td>
HelenaLC
</td>
<td>
2019-03-11
</td>
<td>
initial commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level3">
<h3>Load packages</h3>
<pre class="r"><code>suppressPackageStartupMessages({
    library(cowplot)
    library(ggplot2)
    library(scater)
    library(SingleCellExperiment)
})</code></pre>
<pre><code>Warning: Paket &#39;scater&#39; wurde unter R Version 3.5.2 erstellt</code></pre>
<pre><code>Warning: Paket &#39;SingleCellExperiment&#39; wurde unter R Version 3.5.2 erstellt</code></pre>
<pre><code>Warning: Paket &#39;GenomeInfoDb&#39; wurde unter R Version 3.5.2 erstellt</code></pre>
</div>
<div id="load-reformat-data" class="section level3">
<h3>Load &amp; reformat data</h3>
<pre class="r"><code># load raw counts
fastq_dirs &lt;- list.dirs(&quot;data&quot;, recursive = FALSE, full.names = TRUE)
names(fastq_dirs) &lt;- basename(fastq_dirs)
sce &lt;- DropletUtils::read10xCounts(fastq_dirs)

# rename row/colData &amp; dimnames
names(rowData(sce)) &lt;- c(&quot;feature&quot;, &quot;symbol&quot;)
names(colData(sce)) &lt;- c(&quot;sample_id&quot;, &quot;barcode&quot;)
sce$sample_id &lt;- factor(basename(sce$sample_id))
dimnames(sce) &lt;- list(
    with(rowData(sce), paste(feature, symbol, sep = &quot;.&quot;)),
    with(colData(sce), paste(barcode, sample_id, sep = &quot;.&quot;)))

# load metadata
md_dir &lt;- file.path(&quot;data&quot;, &quot;metadata.xlsx&quot;)
md &lt;- readxl::read_excel(md_dir)
m &lt;- match(sce$sample_id, md$`Sample ID`)
sce$group_id &lt;- md$Characteristics[m]

# remove undetected genes
sce &lt;- sce[Matrix::rowSums(counts(sce) &gt; 0) &gt; 0, ]
dim(sce)</code></pre>
<pre><code>[1] 22963 30185</code></pre>
</div>
<div id="calculate-qc-metrics" class="section level3">
<h3>Calculate QC Metrics</h3>
<pre class="r"><code>(mito &lt;- grep(&quot;mt-&quot;, rownames(sce), value = TRUE))</code></pre>
<pre><code> [1] &quot;ENSMUSG00000064341.mt-Nd1&quot;  &quot;ENSMUSG00000064345.mt-Nd2&quot; 
 [3] &quot;ENSMUSG00000064351.mt-Co1&quot;  &quot;ENSMUSG00000064354.mt-Co2&quot; 
 [5] &quot;ENSMUSG00000064356.mt-Atp8&quot; &quot;ENSMUSG00000064357.mt-Atp6&quot;
 [7] &quot;ENSMUSG00000064358.mt-Co3&quot;  &quot;ENSMUSG00000064360.mt-Nd3&quot; 
 [9] &quot;ENSMUSG00000065947.mt-Nd4l&quot; &quot;ENSMUSG00000064363.mt-Nd4&quot; 
[11] &quot;ENSMUSG00000064367.mt-Nd5&quot;  &quot;ENSMUSG00000064368.mt-Nd6&quot; 
[13] &quot;ENSMUSG00000064370.mt-Cytb&quot;</code></pre>
<pre class="r"><code>sce &lt;- calculateQCMetrics(sce, feature_controls = list(Mt = mito))
plotHighestExprs(sce, n = 20)</code></pre>
<p><img src="figure/0-preprocessing.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="filtering" class="section level3">
<h3>Filtering</h3>
<pre class="r"><code># get sample-specific outliers
cols &lt;- c(&quot;total_counts&quot;, &quot;total_features_by_counts&quot;, &quot;pct_counts_Mt&quot;)
log &lt;- c(TRUE, TRUE, FALSE)
type &lt;- c(&quot;both&quot;, &quot;both&quot;, &quot;higher&quot;)

drop_cols &lt;- paste0(cols, &quot;_drop&quot;)
for (i in seq_along(cols))
    colData(sce)[[drop_cols[i]]] &lt;- isOutlier(sce[[cols[i]]], 
        nmads = 2.5, type = type[i], log = log[i], batch = sce$sample_id)

sapply(drop_cols, function(i) 
    sapply(drop_cols, function(j)
        sum(sce[[i]] &amp; sce[[j]])))</code></pre>
<pre><code>                              total_counts_drop
total_counts_drop                           162
total_features_by_counts_drop                79
pct_counts_Mt_drop                           38
                              total_features_by_counts_drop
total_counts_drop                                        79
total_features_by_counts_drop                            94
pct_counts_Mt_drop                                       31
                              pct_counts_Mt_drop
total_counts_drop                             38
total_features_by_counts_drop                 31
pct_counts_Mt_drop                          3451</code></pre>
<pre class="r"><code>cd &lt;- data.frame(colData(sce))
ps &lt;- lapply(seq_along(cols), function (i) {
    p &lt;- ggplot(cd, aes_string(x = cols[i], alpha = drop_cols[i])) +
        geom_histogram(bins = 100, show.legend = FALSE) +
        scale_alpha_manual(values = c(&quot;FALSE&quot; = 1, &quot;TRUE&quot; = 0.4)) +
        facet_wrap(~sample_id, ncol = 1, scales = &quot;free&quot;) + 
        theme_classic() + theme(strip.background = element_blank())
    if (log[i]) 
        p &lt;- p + scale_x_log10()
    return(p)
})
plot_grid(plotlist = ps, ncol = 3)</code></pre>
<p><img src="figure/0-preprocessing.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ol &lt;- Matrix::rowSums(as.matrix(colData(sce)[drop_cols])) != 0
plot_grid(nrow = 1, labels = c(&quot;unfiltered&quot;, &quot;filtered&quot;),
    LSD::heatscatter(sce$total_counts, sce$total_features_by_counts, 
        xlab = &quot;Total counts&quot;, ylab = &quot;Non-zero features&quot;, log=&quot;xy&quot;),
    LSD::heatscatter(sce$total_counts[ol], sce$total_features_by_counts[ol], 
        xlab = &quot;Total counts&quot;, ylab = &quot;Non-zero features&quot;, log=&quot;xy&quot;))</code></pre>
<p><img src="figure/0-preprocessing.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/0-preprocessing.Rmd/unnamed-chunk-5-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/0-preprocessing.Rmd/unnamed-chunk-5-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># summary of cells kept
ns &lt;- table(sce$sample_id)
ns_fil &lt;- table(sce$sample_id[!ol])
print(rbind(
    unfiltered = ns, filtered = ns_fil, 
    &quot;%&quot; = ns_fil / ns * 100), digits = 0)</code></pre>
<pre><code>           LC016 LC017 LC019 LC020 LC022 LC023 LC025 LC026
unfiltered  4553  2420  2947  4430  3783  3529  3347  5176
filtered    4084  2124  2415  3547  3408  3264  3010  4755
%             90    88    82    80    90    92    90    92</code></pre>
<pre class="r"><code># drop outlier cells
sce &lt;- sce[, !ol]
dim(sce)</code></pre>
<pre><code>[1] 22963 26607</code></pre>
<pre class="r"><code># require count &gt; 1 in at least 20 cells
sce &lt;- sce[Matrix::rowSums(counts(sce) &gt; 1) &gt;= 20, ]
dim(sce)</code></pre>
<pre><code>[1] 11309 26607</code></pre>
</div>
<div id="save-objects" class="section level3">
<h3>Save objects</h3>
<pre class="r"><code>saveRDS(sce, &quot;output/MAGL-SCE_filtered.rds&quot;)</code></pre>
</div>
<div id="session-info" class="section level3">
<h3>Session info</h3>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.14.3

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] C/UTF-8/C/C/C/C

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] scater_1.10.1               SingleCellExperiment_1.4.1 
 [3] SummarizedExperiment_1.12.0 DelayedArray_0.8.0         
 [5] BiocParallel_1.16.6         matrixStats_0.54.0         
 [7] Biobase_2.42.0              GenomicRanges_1.34.0       
 [9] GenomeInfoDb_1.18.2         IRanges_2.16.0             
[11] S4Vectors_0.20.1            BiocGenerics_0.28.0        
[13] cowplot_0.9.4               ggplot2_3.1.0              

loaded via a namespace (and not attached):
 [1] locfit_1.5-9.1           Rcpp_1.0.0              
 [3] lattice_0.20-38          assertthat_0.2.0        
 [5] rprojroot_1.3-2          digest_0.6.18           
 [7] cellranger_1.1.0         R6_2.4.0                
 [9] plyr_1.8.4               backports_1.1.3         
[11] evaluate_0.13            pillar_1.3.1            
[13] zlibbioc_1.28.0          rlang_0.3.1             
[15] readxl_1.3.0             lazyeval_0.2.1          
[17] whisker_0.3-2            Matrix_1.2-16           
[19] rmarkdown_1.11           labeling_0.3            
[21] stringr_1.4.0            RCurl_1.95-4.12         
[23] munsell_0.5.0            HDF5Array_1.10.1        
[25] compiler_3.5.1           vipor_0.4.5             
[27] xfun_0.5                 DropletUtils_1.2.2      
[29] pkgconfig_2.0.2          ggbeeswarm_0.6.0        
[31] htmltools_0.3.6          tidyselect_0.2.5        
[33] tibble_2.0.1             gridExtra_2.3           
[35] GenomeInfoDbData_1.2.0   edgeR_3.24.3            
[37] workflowr_1.2.0          viridisLite_0.3.0       
[39] crayon_1.3.4             dplyr_0.8.0.1           
[41] withr_2.1.2              bitops_1.0-6            
[43] grid_3.5.1               gtable_0.2.0            
[45] git2r_0.24.0             magrittr_1.5            
[47] scales_1.0.0             LSD_4.0-0               
[49] stringi_1.3.1            XVector_0.22.0          
[51] reshape2_1.4.3           viridis_0.5.1           
[53] fs_1.2.6                 limma_3.38.3            
[55] DelayedMatrixStats_1.4.0 Rhdf5lib_1.4.2          
[57] tools_3.5.1              glue_1.3.0              
[59] beeswarm_0.2.3           purrr_0.3.1             
[61] yaml_2.2.0               colorspace_1.4-0        
[63] rhdf5_2.26.2             knitr_1.22              </code></pre>
<br> <br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.14.3

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] C/UTF-8/C/C/C/C

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] scater_1.10.1               SingleCellExperiment_1.4.1 
 [3] SummarizedExperiment_1.12.0 DelayedArray_0.8.0         
 [5] BiocParallel_1.16.6         matrixStats_0.54.0         
 [7] Biobase_2.42.0              GenomicRanges_1.34.0       
 [9] GenomeInfoDb_1.18.2         IRanges_2.16.0             
[11] S4Vectors_0.20.1            BiocGenerics_0.28.0        
[13] cowplot_0.9.4               ggplot2_3.1.0              

loaded via a namespace (and not attached):
 [1] locfit_1.5-9.1           Rcpp_1.0.0              
 [3] lattice_0.20-38          assertthat_0.2.0        
 [5] rprojroot_1.3-2          digest_0.6.18           
 [7] cellranger_1.1.0         R6_2.4.0                
 [9] plyr_1.8.4               backports_1.1.3         
[11] evaluate_0.13            pillar_1.3.1            
[13] zlibbioc_1.28.0          rlang_0.3.1             
[15] readxl_1.3.0             lazyeval_0.2.1          
[17] whisker_0.3-2            Matrix_1.2-16           
[19] rmarkdown_1.11           labeling_0.3            
[21] stringr_1.4.0            RCurl_1.95-4.12         
[23] munsell_0.5.0            HDF5Array_1.10.1        
[25] compiler_3.5.1           vipor_0.4.5             
[27] xfun_0.5                 DropletUtils_1.2.2      
[29] pkgconfig_2.0.2          ggbeeswarm_0.6.0        
[31] htmltools_0.3.6          tidyselect_0.2.5        
[33] tibble_2.0.1             gridExtra_2.3           
[35] GenomeInfoDbData_1.2.0   edgeR_3.24.3            
[37] workflowr_1.2.0          viridisLite_0.3.0       
[39] crayon_1.3.4             dplyr_0.8.0.1           
[41] withr_2.1.2              bitops_1.0-6            
[43] grid_3.5.1               gtable_0.2.0            
[45] git2r_0.24.0             magrittr_1.5            
[47] scales_1.0.0             LSD_4.0-0               
[49] stringi_1.3.1            XVector_0.22.0          
[51] reshape2_1.4.3           viridis_0.5.1           
[53] fs_1.2.6                 limma_3.38.3            
[55] DelayedMatrixStats_1.4.0 Rhdf5lib_1.4.2          
[57] tools_3.5.1              glue_1.3.0              
[59] beeswarm_0.2.3           purrr_0.3.1             
[61] yaml_2.2.0               colorspace_1.4-0        
[63] rhdf5_2.26.2             knitr_1.22              </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
