<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />



<meta name="date" content="2019-07-16" />

<title>Preprocessing</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">LPS</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Contents</a>
</li>
<li>
  <a href="0-preprocessing.html">Preprocessing</a>
</li>
<li>
  <a href="1-clustering.html">Clustering</a>
</li>
<li>
  <a href="2-annotation.html">Annotation</a>
</li>
<li>
  <a href="3-differential.html">DS analysis</a>
</li>
<li>
  <a href="4-geneset.html">Geneset analysis</a>
</li>
<li>
  <a href="5-visualization.html">Visualization</a>
</li>
<li>
  <a href="6-more.html">Downstream</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://robinsonlabuzh.github.io">Robinson Lab</a>
</li>
<li>
  <a href="https://github.com/HelenaLC/muscat-comparison">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Preprocessing</h1>
<h4 class="author">Pierre-Luc Germain</h4>
<address class="author_afil">
Institute for Molecular Life Sciences, University of Zurich, Switzerland<br><h4 class="author">Helena L. Crowell</h4>
<address class="author_afil">
Institute for Molecular Life Sciences, University of Zurich, Switzerland<br><h4 class="date">July 16, 2019</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-07-16
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>MAGL/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.4.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div class="panel-group" id="workflowr-checks">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongobjectspresent"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>Environment:</strong> objects present </a>
</p>
</div>
<div id="strongEnvironmentstrongobjectspresent" class="panel-collapse collapse">
<div class="panel-body">

<p>
The global environment had objects present when the code in the R Markdown file was run. These objects can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment. Use <code>wflow_publish</code> or <code>wflow_build</code> to ensure that the code is always run in an empty environment.
</p>
<p>
The following objects were defined in the global environment when these results were created:
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
Name
</th>
<th style="text-align:left;">
Class
</th>
<th style="text-align:left;">
Size
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
data
</td>
<td style="text-align:left;">
environment
</td>
<td style="text-align:left;">
56 bytes
</td>
</tr>
<tr>
<td style="text-align:left;">
env
</td>
<td style="text-align:left;">
environment
</td>
<td style="text-align:left;">
56 bytes
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190311code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190311)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190311code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190311)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomHelenaLCmuscatcomparisontree282cd5af8ff1a0deca02d3ab2aa29cec0a1a7106targetblank282cd5aa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/HelenaLC/muscat-comparison/tree/282cd5af8ff1a0deca02d3ab2aa29cec0a1a7106" target="_blank">282cd5a</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomHelenaLCmuscatcomparisontree282cd5af8ff1a0deca02d3ab2aa29cec0a1a7106targetblank282cd5aa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    ._.Renviron
    Ignored:    ._.gitignore
    Ignored:    ._README.md
    Ignored:    ._Snakefile
    Ignored:    ._config.yaml
    Ignored:    .snakemake/
    Ignored:    MAGL/.DS_Store
    Ignored:    MAGL/.RData
    Ignored:    MAGL/.Rhistory
    Ignored:    MAGL/._.DS_Store
    Ignored:    MAGL/._.Rprofile
    Ignored:    MAGL/._.gitignore
    Ignored:    MAGL/._data
    Ignored:    MAGL/analysis/._refs.bib
    Ignored:    MAGL/analysis/1-clustering_cache/
    Ignored:    MAGL/analysis/3-differential_cache/
    Ignored:    MAGL/analysis/4-visualization_cache/
    Ignored:    MAGL/analysis/5-geneset_cache/
    Ignored:    MAGL/analysis/6-more_cache/
    Ignored:    MAGL/data/
    Ignored:    MAGL/output/
    Ignored:    data/
    Ignored:    figures/
    Ignored:    meta/
    Ignored:    results/
    Ignored:    scripts/.DS_Store
    Ignored:    scripts/._.DS_Store
    Ignored:    scripts/._apply_scdd.R
    Ignored:    scripts/._plot_perf_by_expr.R
    Ignored:    scripts/._plot_perf_by_lfc.R
    Ignored:    scripts/._plot_perf_by_ss.R
    Ignored:    scripts/._plot_runtimes.R
    Ignored:    scripts/._plot_sim_ex.R
    Ignored:    scripts/._prep_data.R
    Ignored:    scripts/._prep_magl.R
    Ignored:    scripts/._prep_sim.R
    Ignored:    scripts/._sim_qc.R
    Ignored:    scripts/._utils.R

Untracked files:
    Untracked:  .RData
    Untracked:  .RDataTmp
    Untracked:  .Rhistory
    Untracked:  MAGL/current
    Untracked:  figs/

Unstaged changes:
    Modified:   MAGL/.Rprofile
    Modified:   MAGL/analysis/3-differential.Rmd
    Deleted:    scripts/fig_vehicle_vs_lps.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/HelenaLC/muscat-comparison/blob/244411f3466962bb727005a8b10b41349a588742/MAGL/analysis/0-preprocessing.Rmd" target="_blank">244411f</a>
</td>
<td>
HelenaLC
</td>
<td>
2019-07-11
</td>
<td>
update lps analysis to include removal of doublets
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/HelenaLC/muscat-comparison/244411f3466962bb727005a8b10b41349a588742/MAGL/docs/0-preprocessing.html" target="_blank">244411f</a>
</td>
<td>
HelenaLC
</td>
<td>
2019-07-11
</td>
<td>
update lps analysis to include removal of doublets
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/HelenaLC/muscat-comparison/08bf26003c87dc52a3395028d1117b7a0ffb75a9/MAGL/docs/0-preprocessing.html" target="_blank">08bf260</a>
</td>
<td>
HelenaLC
</td>
<td>
2019-06-17
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/HelenaLC/muscat-comparison/blob/61c0b067f9144eecf49366d6d9bc4000c4d10986/MAGL/analysis/0-preprocessing.Rmd" target="_blank">61c0b06</a>
</td>
<td>
HelenaLC
</td>
<td>
2019-05-20
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/HelenaLC/muscat-comparison/blob/faf858d01b3ab307553b2e18a712af44215401d1/MAGL/analysis/0-preprocessing.Rmd" target="_blank">faf858d</a>
</td>
<td>
HelenaLC
</td>
<td>
2019-05-17
</td>
<td>
add DS &amp; geneset analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/HelenaLC/muscat-comparison/3c5aa5f379847c31386b94463c65fa6295f26fc2/MAGL/docs/0-preprocessing.html" target="_blank">3c5aa5f</a>
</td>
<td>
HelenaLC
</td>
<td>
2019-05-03
</td>
<td>
add MAGL prepro,clust,anno
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/HelenaLC/muscat-comparison/blob/10c7525a571656c0b7abe7bc179a09c3d477149b/MAGL/analysis/0-preprocessing.Rmd" target="_blank">10c7525</a>
</td>
<td>
HelenaLC
</td>
<td>
2019-05-03
</td>
<td>
add MAGL prepro,clust,anno
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/HelenaLC/muscat-comparison/ba6cb8c10455e4e867de937087d2d2831b0a1881/MAGL/docs/0-preprocessing.html" target="_blank">ba6cb8c</a>
</td>
<td>
HelenaLC
</td>
<td>
2019-04-03
</td>
<td>
add scripts
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/HelenaLC/muscat-comparison/blob/eaed7ecf04d1ff5ca101c68dfa7079d114b49658/MAGL/analysis/0-preprocessing.Rmd" target="_blank">eaed7ec</a>
</td>
<td>
HelenaLC
</td>
<td>
2019-03-11
</td>
<td>
initial commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level3">
<h3>Load packages</h3>
<pre class="r"><code>library(cowplot)
library(ggplot2)
library(scater)
library(scds)
library(SingleCellExperiment)</code></pre>
</div>
<div id="load-reformat-data" class="section level1">
<h1>Load &amp; reformat data</h1>
<pre class="r"><code># load raw counts
fastq_dirs &lt;- list.dirs(&quot;data&quot;, recursive = FALSE, full.names = TRUE)
names(fastq_dirs) &lt;- basename(fastq_dirs)
sce &lt;- DropletUtils::read10xCounts(fastq_dirs)

# rename row/colData colnames &amp; SCE dimnames
names(rowData(sce)) &lt;- c(&quot;ENSEMBL&quot;, &quot;SYMBOL&quot;)
names(colData(sce)) &lt;- c(&quot;sample_id&quot;, &quot;barcode&quot;)
sce$sample_id &lt;- factor(basename(sce$sample_id))
dimnames(sce) &lt;- list(
    with(rowData(sce), paste(ENSEMBL, SYMBOL, sep = &quot;.&quot;)),
    with(colData(sce), paste(barcode, sample_id, sep = &quot;.&quot;)))

# load metadata
md_dir &lt;- file.path(&quot;data&quot;, &quot;metadata.xlsx&quot;)
md &lt;- readxl::read_excel(md_dir)
m &lt;- match(sce$sample_id, md$`Sample ID`)
sce$group_id &lt;- md$Characteristics[m]

# remove undetected genes
sce &lt;- sce[Matrix::rowSums(counts(sce) &gt; 0) &gt; 0, ]
dim(sce)</code></pre>
<pre><code>[1] 22963 30185</code></pre>
</div>
<div id="doublet-removal" class="section level1">
<h1>Doublet removal</h1>
<pre class="r"><code># split SCE by sample
cs_by_s &lt;- split(colnames(sce), sce$sample_id)
sce_by_s &lt;- lapply(cs_by_s, function(cs) sce[, cs])

# run &#39;scds&#39;
sce_by_s &lt;- lapply(sce_by_s, function(u) 
    cxds_bcds_hybrid(bcds(cxds(u))))

# remove doublets
sce_by_s &lt;- lapply(sce_by_s, function(u) {
    # compute expected nb. of doublets (10x)
    n_dbl &lt;- ceiling(0.01 * ncol(u)^2 / 1e3)
    # remove &#39;n_dbl&#39; cells w/ highest doublet score
    o &lt;- order(u$hybrid_score, decreasing = TRUE)
    u[, -o[seq_len(n_dbl)]]
})

# merge back into single SCE
sce &lt;- do.call(cbind, sce_by_s)</code></pre>
</div>
<div id="calculate-qc-metrics" class="section level1">
<h1>Calculate QC Metrics</h1>
<pre class="r"><code>(mito &lt;- grep(&quot;mt-&quot;, rownames(sce), value = TRUE))</code></pre>
<pre><code> [1] &quot;ENSMUSG00000064341.mt-Nd1&quot;  &quot;ENSMUSG00000064345.mt-Nd2&quot; 
 [3] &quot;ENSMUSG00000064351.mt-Co1&quot;  &quot;ENSMUSG00000064354.mt-Co2&quot; 
 [5] &quot;ENSMUSG00000064356.mt-Atp8&quot; &quot;ENSMUSG00000064357.mt-Atp6&quot;
 [7] &quot;ENSMUSG00000064358.mt-Co3&quot;  &quot;ENSMUSG00000064360.mt-Nd3&quot; 
 [9] &quot;ENSMUSG00000065947.mt-Nd4l&quot; &quot;ENSMUSG00000064363.mt-Nd4&quot; 
[11] &quot;ENSMUSG00000064367.mt-Nd5&quot;  &quot;ENSMUSG00000064368.mt-Nd6&quot; 
[13] &quot;ENSMUSG00000064370.mt-Cytb&quot;</code></pre>
<pre class="r"><code>sce &lt;- calculateQCMetrics(sce, feature_controls = list(Mt = mito))
plotHighestExprs(sce, n = 20)</code></pre>
<p><img src="figure/0-preprocessing.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/HelenaLC/muscat-comparison/blob/244411f3466962bb727005a8b10b41349a588742/MAGL/docs/figure/0-preprocessing.Rmd/unnamed-chunk-2-1.png" target="_blank">244411f</a>
</td>
<td>
HelenaLC
</td>
<td>
2019-07-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="filtering" class="section level1">
<h1>Filtering</h1>
<pre class="r"><code># get sample-specific outliers
cols &lt;- c(&quot;total_counts&quot;, &quot;total_features_by_counts&quot;, &quot;pct_counts_Mt&quot;)
log &lt;- c(TRUE, TRUE, FALSE)
type &lt;- c(&quot;both&quot;, &quot;both&quot;, &quot;higher&quot;)

drop_cols &lt;- paste0(cols, &quot;_drop&quot;)
for (i in seq_along(cols))
    colData(sce)[[drop_cols[i]]] &lt;- isOutlier(sce[[cols[i]]], 
        nmads = 2.5, type = type[i], log = log[i], batch = sce$sample_id)

sapply(drop_cols, function(i) 
    sapply(drop_cols, function(j)
        sum(sce[[i]] &amp; sce[[j]])))</code></pre>
<pre><code>                              total_counts_drop
total_counts_drop                           101
total_features_by_counts_drop                50
pct_counts_Mt_drop                           20
                              total_features_by_counts_drop
total_counts_drop                                        50
total_features_by_counts_drop                            68
pct_counts_Mt_drop                                       23
                              pct_counts_Mt_drop
total_counts_drop                             20
total_features_by_counts_drop                 23
pct_counts_Mt_drop                          3341</code></pre>
<pre class="r"><code>cd &lt;- data.frame(colData(sce))
ps &lt;- lapply(seq_along(cols), function (i) {
    p &lt;- ggplot(cd, aes_string(x = cols[i], alpha = drop_cols[i])) +
        geom_histogram(bins = 100, show.legend = FALSE) +
        scale_alpha_manual(values = c(&quot;FALSE&quot; = 1, &quot;TRUE&quot; = 0.4)) +
        facet_wrap(~sample_id, ncol = 1, scales = &quot;free&quot;) + 
        theme_classic() + theme(strip.background = element_blank())
    if (log[i]) 
        p &lt;- p + scale_x_log10()
    return(p)
})
plot_grid(plotlist = ps, ncol = 3)</code></pre>
<p><img src="figure/0-preprocessing.Rmd/unnamed-chunk-3-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/HelenaLC/muscat-comparison/blob/244411f3466962bb727005a8b10b41349a588742/MAGL/docs/figure/0-preprocessing.Rmd/unnamed-chunk-3-1.png" target="_blank">244411f</a>
</td>
<td>
HelenaLC
</td>
<td>
2019-07-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/HelenaLC/muscat-comparison/blob/67f5a1b688156232aa3ee82b60730aecf6f7757e/MAGL/docs/figure/0-preprocessing.Rmd/unnamed-chunk-3-1.png" target="_blank">67f5a1b</a>
</td>
<td>
HelenaLC
</td>
<td>
2019-05-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/HelenaLC/muscat-comparison/blob/ba6cb8c10455e4e867de937087d2d2831b0a1881/MAGL/docs/figure/0-preprocessing.Rmd/unnamed-chunk-3-1.png" target="_blank">ba6cb8c</a>
</td>
<td>
HelenaLC
</td>
<td>
2019-04-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>layout(matrix(1:2, nrow = 1))
ol &lt;- Matrix::rowSums(as.matrix(colData(sce)[drop_cols])) != 0
x &lt;- sce$total_counts
y &lt;- sce$total_features_by_counts
LSD::heatscatter(x, y, log=&quot;xy&quot;, main = &quot;unfiltered&quot;, 
    xlab = &quot;Total counts&quot;, ylab = &quot;Non-zero features&quot;)
LSD::heatscatter(x[!ol], y[!ol], log=&quot;xy&quot;, main = &quot;filtered&quot;, 
    xlab = &quot;Total counts&quot;, ylab = &quot;Non-zero features&quot;)</code></pre>
<p><img src="figure/0-preprocessing.Rmd/unnamed-chunk-4-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/HelenaLC/muscat-comparison/blob/244411f3466962bb727005a8b10b41349a588742/MAGL/docs/figure/0-preprocessing.Rmd/unnamed-chunk-4-1.png" target="_blank">244411f</a>
</td>
<td>
HelenaLC
</td>
<td>
2019-07-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/HelenaLC/muscat-comparison/blob/08bf26003c87dc52a3395028d1117b7a0ffb75a9/MAGL/docs/figure/0-preprocessing.Rmd/unnamed-chunk-4-1.png" target="_blank">08bf260</a>
</td>
<td>
HelenaLC
</td>
<td>
2019-06-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/HelenaLC/muscat-comparison/blob/67f5a1b688156232aa3ee82b60730aecf6f7757e/MAGL/docs/figure/0-preprocessing.Rmd/unnamed-chunk-4-1.png" target="_blank">67f5a1b</a>
</td>
<td>
HelenaLC
</td>
<td>
2019-05-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/HelenaLC/muscat-comparison/blob/ba6cb8c10455e4e867de937087d2d2831b0a1881/MAGL/docs/figure/0-preprocessing.Rmd/unnamed-chunk-4-1.png" target="_blank">ba6cb8c</a>
</td>
<td>
HelenaLC
</td>
<td>
2019-04-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># summary of cells kept
ns &lt;- table(sce$sample_id)
ns_fil &lt;- table(sce$sample_id[!ol])
print(rbind(
    unfiltered = ns, filtered = ns_fil, 
    &quot;%&quot; = ns_fil / ns * 100), digits = 0)</code></pre>
<pre><code>           LC016 LC017 LC019 LC020 LC022 LC023 LC025 LC026
unfiltered  4345  2361  2860  4233  3639  3404  3234  4908
filtered    3899  2074  2342  3375  3273  3158  2926  4510
%             90    88    82    80    90    93    90    92</code></pre>
<pre class="r"><code># drop outlier cells
sce &lt;- sce[, !ol]
dim(sce)</code></pre>
<pre><code>[1] 22963 25557</code></pre>
<pre class="r"><code># require count &gt; 1 in at least 20 cells
sce &lt;- sce[Matrix::rowSums(counts(sce) &gt; 1) &gt;= 20, ]
dim(sce)</code></pre>
<pre><code>[1] 11063 25557</code></pre>
</div>
<div id="save-sce-to-.rds" class="section level1">
<h1>Save SCE to .rds</h1>
<pre class="r"><code>saveRDS(sce, file.path(&quot;output&quot;, &quot;MAGL-SCE.rds&quot;))</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.0 (2019-04-26)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 16.04.5 LTS

Matrix products: default
BLAS:   /usr/local/R/R-3.6.0/lib/libRblas.so
LAPACK: /usr/local/R/R-3.6.0/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_CA.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_CA.UTF-8        LC_COLLATE=en_CA.UTF-8    
 [5] LC_MONETARY=en_CA.UTF-8    LC_MESSAGES=en_CA.UTF-8   
 [7] LC_PAPER=en_CA.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_CA.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] scds_1.0.0                  scater_1.12.2              
 [3] SingleCellExperiment_1.6.0  SummarizedExperiment_1.14.0
 [5] DelayedArray_0.10.0         BiocParallel_1.18.0        
 [7] matrixStats_0.54.0          Biobase_2.44.0             
 [9] GenomicRanges_1.36.0        GenomeInfoDb_1.20.0        
[11] IRanges_2.18.1              S4Vectors_0.22.0           
[13] BiocGenerics_0.30.0         cowplot_0.9.4              
[15] ggplot2_3.2.0              

loaded via a namespace (and not attached):
 [1] viridis_0.5.1            edgeR_3.26.5            
 [3] BiocSingular_1.0.0       viridisLite_0.3.0       
 [5] R.utils_2.9.0            DelayedMatrixStats_1.6.0
 [7] assertthat_0.2.1         highr_0.8               
 [9] dqrng_0.2.1              cellranger_1.1.0        
[11] GenomeInfoDbData_1.2.1   vipor_0.4.5             
[13] yaml_2.2.0               pillar_1.4.2            
[15] backports_1.1.4          lattice_0.20-38         
[17] limma_3.40.2             glue_1.3.1              
[19] digest_0.6.20            XVector_0.24.0          
[21] colorspace_1.4-1         R.oo_1.22.0             
[23] htmltools_0.3.6          Matrix_1.2-17           
[25] pkgconfig_2.0.2          zlibbioc_1.30.0         
[27] purrr_0.3.2              scales_1.0.0            
[29] HDF5Array_1.12.1         whisker_0.3-2           
[31] LSD_4.0-0                git2r_0.26.1            
[33] tibble_2.1.3             xgboost_0.82.1          
[35] withr_2.1.2              lazyeval_0.2.2          
[37] readxl_1.3.1             magrittr_1.5            
[39] crayon_1.3.4             evaluate_0.14           
[41] R.methodsS3_1.7.1        fs_1.3.1                
[43] beeswarm_0.2.3           tools_3.6.0             
[45] data.table_1.12.2        stringr_1.4.0           
[47] Rhdf5lib_1.6.0           DropletUtils_1.4.2      
[49] locfit_1.5-9.1           munsell_0.5.0           
[51] irlba_2.3.3              compiler_3.6.0          
[53] rsvd_1.0.1               rlang_0.4.0             
[55] rhdf5_2.28.0             grid_3.6.0              
[57] RCurl_1.95-4.12          BiocNeighbors_1.2.0     
[59] labeling_0.3             bitops_1.0-6            
[61] rmarkdown_1.13           gtable_0.3.0            
[63] codetools_0.2-16         R6_2.4.0                
[65] gridExtra_2.3            knitr_1.23              
[67] dplyr_0.8.3              workflowr_1.4.0         
[69] rprojroot_1.3-2          stringi_1.4.3           
[71] ggbeeswarm_0.6.0         Rcpp_1.0.1              
[73] tidyselect_0.2.5         xfun_0.8                </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
