<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />



<meta name="date" content="2019-05-15" />

<title>DS analysis</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">MAGL</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="0-preprocessing.html">Preprocessing</a>
</li>
<li>
  <a href="1-clustering.html">Clustering</a>
</li>
<li>
  <a href="2-annotation.html">Annotation</a>
</li>
<li>
  <a href="3-differential.html">DS analysis</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://robinsonlabuzh.github.io">Robinson Lab</a>
</li>
<li>
  <a href="https://github.com/HelenaLC/muscat-comparison">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">DS analysis</h1>
<h4 class="author">Helena L. Crowell</h4>
<address class="author_afil">
Institute for Molecular Life Sciences, University of Zurich, Switzerland<br><h4 class="date">May 15, 2019</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-05-15
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 4 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>MAGL/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.3.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190311code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190311)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190311code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190311)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongdetected"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>Cache:</strong> detected </a>
</p>
</div>
<div id="strongCachestrongdetected" class="panel-collapse collapse">
<div class="panel-body">
The following chunks had caches available:
<ul>
<li>
= FALSE
</li>
<li>
agg
</li>
<li>
cluster-anno
</li>
<li>
data
</li>
<li>
ds-analysis
</li>
<li>
pb-mds
</li>
<li>
prep-data
</li>
<li>
unnamed-chunk-2
</li>
<li>
unnamed-chunk-3
</li>
<li>
violins
</li>
</ul>
<p>To ensure reproducibility of the results, delete the cache directory <code>3-differential_cache</code> and re-run the analysis. To have workflowr automatically delete the cache directory prior to building the file, set <code>delete_cache = TRUE</code> when running <code>wflow_build()</code> or <code>wflow_publish()</code>.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomHelenaLCmuscatcomparisontree67f5a1b688156232aa3ee82b60730aecf6f7757etargetblank67f5a1ba"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/HelenaLC/muscat-comparison/tree/67f5a1b688156232aa3ee82b60730aecf6f7757e" target="_blank">67f5a1b</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomHelenaLCmuscatcomparisontree67f5a1b688156232aa3ee82b60730aecf6f7757etargetblank67f5a1ba" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .snakemake/
    Ignored:    MAGL/.DS_Store
    Ignored:    MAGL/.RData
    Ignored:    MAGL/.Rhistory
    Ignored:    MAGL/analysis/3-differential_cache/
    Ignored:    MAGL/analysis/figure/
    Ignored:    MAGL/data/
    Ignored:    MAGL/output/
    Ignored:    config/
    Ignored:    data/
    Ignored:    figures/
    Ignored:    metadata/
    Ignored:    results/

Untracked files:
    Untracked:  .RData
    Untracked:  .Rhistory
    Untracked:  ._README.md
    Untracked:  ._Snakefile
    Untracked:  ._config.yaml
    Untracked:  MAGL/._.DS_Store
    Untracked:  MAGL/._.Rprofile
    Untracked:  MAGL/._data
    Untracked:  MAGL/analysis/._refs.bib
    Untracked:  MAGL/analysis/3-differential.Rmd
    Untracked:  MAGL/analysis/refs.bib
    Untracked:  MAGL/docs/figure/3-differential.Rmd/
    Untracked:  scripts/._apply_ad.R
    Untracked:  scripts/._apply_mm.R
    Untracked:  scripts/._apply_scdd.R
    Untracked:  scripts/._calc_conc.R
    Untracked:  scripts/._countsimQC.R
    Untracked:  scripts/._data_sim.R
    Untracked:  scripts/._meth_pars.R
    Untracked:  scripts/._plot_conc.R
    Untracked:  scripts/._plot_padj_local_vs_global.R
    Untracked:  scripts/._plot_pbMDS.R
    Untracked:  scripts/._plot_perf_combined.R
    Untracked:  scripts/._plot_perf_split_by_cat.R
    Untracked:  scripts/._plot_perf_split_by_lfc.R
    Untracked:  scripts/._plot_perf_vs_ncells.R
    Untracked:  scripts/._plot_sim_pars.R
    Untracked:  scripts/._plot_summary.R
    Untracked:  scripts/._plot_upsets.R
    Untracked:  scripts/._run_method.R
    Untracked:  scripts/._sim_data.R
    Untracked:  scripts/._sim_pars.R
    Untracked:  scripts/._utils.R
    Untracked:  scripts/plot_padj_local_vs_global.R
    Untracked:  scripts/plot_pbMDS.R
    Untracked:  scripts/plot_perf_combined.R
    Untracked:  scripts/plot_perf_split_by_cat.R
    Untracked:  scripts/plot_perf_split_by_lfc.R
    Untracked:  scripts/plot_sim_ex_hm.R
    Untracked:  scripts/plot_sim_pars.R
    Untracked:  scripts/plot_sim_vs_est_logFC.R
    Untracked:  scripts/plot_summary.R

Unstaged changes:
    Modified:   MAGL/analysis/2-annotation.Rmd
    Modified:   MAGL/analysis/_site.yml
    Deleted:    MAGL/analysis/about.Rmd
    Modified:   MAGL/analysis/index.Rmd
    Deleted:    MAGL/analysis/license.Rmd
    Modified:   Snakefile
    Modified:   config.yaml
    Modified:   scripts/apply_ad.R
    Modified:   scripts/apply_mast.R
    Modified:   scripts/apply_mm.R
    Modified:   scripts/apply_pb.R
    Modified:   scripts/apply_scdd.R
    Modified:   scripts/calc_conc.R
    Modified:   scripts/countsimQC.R
    Modified:   scripts/meth_pars.R
    Modified:   scripts/plot_conc.R
    Modified:   scripts/plot_null.R
    Deleted:    scripts/plot_padj_loc_vs_glb.R
    Modified:   scripts/plot_pb_mean_vs_disp.R
    Modified:   scripts/plot_perf.R
    Deleted:    scripts/plot_perf_split.R
    Modified:   scripts/plot_perf_vs_ncells.R
    Deleted:    scripts/plot_perf_vs_ncells_box.R
    Modified:   scripts/plot_upsets.R
    Modified:   scripts/run_method.R
    Modified:   scripts/sim_data.R
    Modified:   scripts/sim_pars.R
    Modified:   scripts/utils.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level3">
<h3>load packages</h3>
<pre class="r"><code>library(ComplexHeatmap)
library(cowplot)
library(limma)
library(magrittr)
library(muscat)
library(scater)
library(SingleCellExperiment)</code></pre>
</div>
<div id="data-preparation" class="section level1">
<h1>Data preparation</h1>
<div id="load-data-prep.-for-muscat" class="section level2">
<h2>Load data &amp; prep. for <code>muscat</code></h2>
<p><code>drop = FALSE</code> specifies that we wish to retain all available cell metadata columns besides those specified as <code>group/sample/cluster_id</code> columns.</p>
<pre class="r"><code>sce0 &lt;- readRDS(file.path(&quot;output&quot;, &quot;MAGL-SCE.rds&quot;))

# concat. sample &amp; group ID
sce0$id &lt;- factor(paste(sce0$sample_id, sce0$group_id, sep = &quot;_&quot;))
# make WT reference group
sce0$group_id &lt;- factor(sce0$group_id, levels = c(&quot;WT&quot;, &quot;LPS&quot;))
# reorder sample levels
o &lt;- sapply(levels(sce0$group_id), grep, levels(sce0$id))
sce0$id &lt;- factor(sce0$id, levels = levels(sce0$id)[o])

sce &lt;- prepSCE(sce0,
    group_id = &quot;group_id&quot;,
    sample_id = &quot;id&quot;,
    cluster_id = &quot;integrated_snn_res.0.2&quot;,
    drop = FALSE)

metadata(sce)$experiment_info</code></pre>
<pre><code>  sample_id group_id n_cells
1  LC016_WT       WT    4084
2  LC019_WT       WT    2415
3  LC022_WT       WT    3408
4  LC025_WT       WT    3010
5 LC017_LPS      LPS    2124
6 LC020_LPS      LPS    3547
7 LC023_LPS      LPS    3264
8 LC026_LPS      LPS    4755</code></pre>
</div>
<div id="cluster-annotation" class="section level2">
<h2>Cluster annotation</h2>
<p>In consideration of the visualizations provided in the <a href="docs/2-annotation.html">Annotation</a> tab and additional exploration with <em><a href="https://bioconductor.org/packages/3.8/iSEE">iSEE</a></em>, we arrive at the following cluster annotations:</p>
<pre class="r"><code>anno &lt;- list(
    &quot;excit_neuron1&quot; = c(0, 1, 4, 5, 16),
    &quot;excit_neuron2&quot; = 7,
    &quot;excit_neuron3&quot; = c(2, 20),
    &quot;excit_neuron4&quot; = 9,
    &quot;excit_neuron5&quot; = 14,
    &quot;excit_neuron6&quot; = 8,
    &quot;inhib_neuron1&quot; = c(10, 15),
    &quot;inhib_neuron2&quot; = 11,
    &quot;inhib_neuron3&quot; = 12,
    &quot;Astrocytes&quot; = 3,
    &quot;OPC&quot; = 13,
    &quot;Oligodendrocytes&quot; = 6,
    &quot;Endothelial&quot; = 18,
    &quot;Microglia&quot; = 19)
# unassigned: 17, 21, 22, 23

m &lt;- match(sce$cluster_id, unlist(anno))
ns &lt;- vapply(anno, length, numeric(1))
lab &lt;- rep.int(names(anno), ns)
sce$cluster_id &lt;- lab[m]

# remove unassigned cells
sce &lt;- sce[, !is.na(sce$cluster_id)]
sce$cluster_id &lt;- factor(sce$cluster_id)
dim(sce)</code></pre>
<pre><code>[1] 11309 25613</code></pre>
</div>
</div>
<div id="aggregation-of-single-cell-to-pseudo-bulk-data" class="section level1">
<h1>Aggregation of single-cell to pseudo-bulk data</h1>
<pre class="r"><code>system.time(pb &lt;- aggregateData(sce))</code></pre>
<pre><code>   user  system elapsed 
 24.804   1.416  15.274 </code></pre>
<pre class="r"><code># SCE of dim. #(genes) x #(samples)
pb </code></pre>
<pre><code>class: SingleCellExperiment 
dim: 11309 8 
metadata(2): experiment_info agg_pars
assays(14): Astrocytes Endothelial ... Oligodendrocytes OPC
rownames(11309): ENSMUSG00000051951.Xkr4 ENSMUSG00000089699.Gm1992
  ... ENSMUSG00000095041.PISD ENSMUSG00000063897.DHRSX
rowData names(0):
colnames(8): LC016_WT LC019_WT ... LC023_LPS LC026_LPS
colData names(0):
reducedDimNames(0):
spikeNames(0):</code></pre>
<pre class="r"><code># one assay per cluster
assayNames(pb) </code></pre>
<pre><code> [1] &quot;Astrocytes&quot;       &quot;Endothelial&quot;      &quot;excit_neuron1&quot;   
 [4] &quot;excit_neuron2&quot;    &quot;excit_neuron3&quot;    &quot;excit_neuron4&quot;   
 [7] &quot;excit_neuron5&quot;    &quot;excit_neuron6&quot;    &quot;inhib_neuron1&quot;   
[10] &quot;inhib_neuron2&quot;    &quot;inhib_neuron3&quot;    &quot;Microglia&quot;       
[13] &quot;Oligodendrocytes&quot; &quot;OPC&quot;             </code></pre>
<pre class="r"><code># view 1st cluster
head(assay(pb))</code></pre>
<pre><code>                           LC016_WT LC019_WT LC022_WT LC025_WT LC017_LPS
ENSMUSG00000051951.Xkr4           7       11       13       30         3
ENSMUSG00000089699.Gm1992         2        2        2        3         0
ENSMUSG00000033845.Mrpl15         5       18       21       16         1
ENSMUSG00000033813.Tcea1          5       17       16       10         6
ENSMUSG00000002459.Rgs20         33      275      219       94        82
ENSMUSG00000033793.Atp6v1h        7       25       20       17         4
                           LC020_LPS LC023_LPS LC026_LPS
ENSMUSG00000051951.Xkr4            7         9        19
ENSMUSG00000089699.Gm1992          0         1         2
ENSMUSG00000033845.Mrpl15         13         5        25
ENSMUSG00000033813.Tcea1          20         9        25
ENSMUSG00000002459.Rgs20         344       217       413
ENSMUSG00000033793.Atp6v1h        36        21        42</code></pre>
<div id="pseudobulk-level-mds-plot" class="section level2">
<h2>Pseudobulk-level MDS plot</h2>
<p>Prior to conducting any formal testing, we can compute a multi-dimensional scaling (MDS) plot of aggregated signal to explore overall sample similarities. Ideally, such a represenation of the data should separate both clusters and groups from one another. Vice versa, samples from the same cluster/group should fall close to each other.</p>
<p>In our MDS plot on pseudobulk counts (Fig. @ref(fig:pb-mds)), we can observe that cell-populations (clusters) are separated quite well, while WT and stimulated samples (groups) are separated to a lesser (if any) extend.</p>
<pre class="r"><code>pbMDS(pb)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/3-differential.Rmd/pb-mds-1.png" alt="Pseudobulk-level MDS plot. Each point represent one cluster-sample instance; points are colored by cluster ID and shaped by group ID." width="672" />
<p class="caption">
Pseudobulk-level MDS plot. Each point represent one cluster-sample instance; points are colored by cluster ID and shaped by group ID.
</p>
</div>
</div>
</div>
<div id="ds-analysis" class="section level1">
<h1>DS analysis</h1>
<pre class="r"><code># specify design &amp; contrast matrix
ei &lt;- metadata(sce)$experiment_info
(design &lt;- model.matrix(~ 0 + ei$group_id) %&gt;% 
    set_rownames(levels(ei$sample_id)) %&gt;% 
    set_colnames(levels(ei$group_id)))</code></pre>
<pre><code>          WT LPS
LC016_WT   1   0
LC017_LPS  1   0
LC019_WT   1   0
LC020_LPS  1   0
LC022_WT   0   1
LC023_LPS  0   1
LC025_WT   0   1
LC026_LPS  0   1
attr(,&quot;assign&quot;)
[1] 1 1
attr(,&quot;contrasts&quot;)
attr(,&quot;contrasts&quot;)$`ei$group_id`
[1] &quot;contr.treatment&quot;</code></pre>
<pre class="r"><code>(contrast &lt;- makeContrasts(&quot;LPS-WT&quot;, levels = design))</code></pre>
<pre><code>      Contrasts
Levels LPS-WT
   WT      -1
   LPS      1</code></pre>
<pre class="r"><code>system.time(res &lt;- pbDS(sce, pb, design, contrast, verbose = FALSE))</code></pre>
<pre><code>   user  system elapsed 
 47.932   0.900  48.877 </code></pre>
<div id="results-filtering-overview" class="section level2">
<h2>Results filtering &amp; overview</h2>
<p>To get a general overview of the differential testing results, we first filter them to retain hits with <span class="math inline">\(\text{FDR}&lt;5\%\)</span> and <span class="math inline">\(~|~logFC~|~&gt;1\)</span>, and view the number &amp; percentage of differential findings by cluster. Finally, we extract the top hits (lowest adj. p-value) in each cluster.</p>
<pre class="r"><code>tbl0 &lt;- res$table$`LPS-WT`
# filter abs(logFC) &gt; 1 &amp; FDR &lt; 0.05
tbl &lt;- lapply(tbl0, dplyr::filter, abs(logFC) &gt; 1, p_adj.loc &lt; 0.05)
# sort by FDR
tbl &lt;- lapply(tbl, dplyr::arrange, p_adj.loc)

# nb. &amp; % of hits by cluster
n_de &lt;- vapply(tbl, nrow, numeric(1))
cbind(n_de, perc = round(n_de / nrow(sce) * 100, 2))</code></pre>
<pre><code>                 n_de perc
Astrocytes          0 0.00
Endothelial         0 0.00
excit_neuron1       0 0.00
excit_neuron2       3 0.03
excit_neuron3       2 0.02
excit_neuron4       0 0.00
excit_neuron5       4 0.04
excit_neuron6      18 0.16
inhib_neuron1       0 0.00
inhib_neuron2       2 0.02
inhib_neuron3       6 0.05
Microglia           0 0.00
Oligodendrocytes    4 0.04
OPC                 2 0.02</code></pre>
<pre class="r"><code># get top-hits for ea. cluster
top_gs &lt;- lapply(tbl, function(u) {
    n &lt;- ifelse(nrow(u) &lt; 6, nrow(u), 6)
    u$gene[seq_len(n)]
})</code></pre>
</div>
</div>
<div id="visualizing-results" class="section level1">
<h1>Visualizing results</h1>
<div id="cell-level-visualization" class="section level2">
<h2>Cell-level visualization</h2>
<p>The <em><a href="https://bioconductor.org/packages/3.8/scater">scater</a></em><span class="citation">(McCarthy et al. 2017)</span> packages provides a variety of visualizations for single-cell data. Here, we use <code>plotExpression</code> to render violin plots of the top differential genes identified for each cluster. We specify <code>x = &quot;sample_id&quot;</code> to obtain one violin per sample, and <code>colour_by = &quot;group_id&quot;</code> to signify the experimental condition each sample belongs to.</p>
<pre class="r"><code># split cells by cluster
cs_by_k &lt;- split(colnames(sce), sce$cluster_id)

kids &lt;- levels(sce$cluster_id)
names(kids) &lt;- kids

ps &lt;- lapply(kids, function(k)
    plotExpression(sce[, cs_by_k[[k]]], features = top_gs[[k]], 
        x = &quot;sample_id&quot;, colour_by = &quot;group_id&quot;) +
        guides(fill = guide_legend(override.aes = list(size = 2, alpha = 1))) +
        theme_classic() + theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1))) </code></pre>
<pre class="echo"><code>for (k in kids) {
  cat(&quot;##### &quot;,k, &quot;\n&quot;)
  print(ps[[k]])
  cat(&quot;\n\n&quot;)
}</code></pre>
</div>
<div id="sample-level-visualization" class="section level2">
<h2>Sample-level visualization</h2>
<pre class="r"><code># normalize &amp; compute cluster-sample expression-means
sce &lt;- normalize(sce)
ms &lt;- aggregateData(sce, assay = &quot;logcounts&quot;, fun = &quot;mean&quot;)

# wrapper for z-normalization
.z_norm &lt;- function(x, th = 2.5) {
    x &lt;- as.matrix(x)
    sds &lt;- rowSds(x, na.rm = TRUE)
    sds[sds == 0] &lt;- 1
    x &lt;- t(t(x - rowMeans(x, na.rm = TRUE)) / sds)
    #x &lt;- (x - rowMeans(x, na.rm = TRUE)) / sds
    x[x &gt;  th] &lt;-  th
    x[x &lt; -th] &lt;- -th
    return(x)
}

# wrapper to plot differential heatmap
.plot_diff_hm &lt;- function(ms, k, gs, ei) {
    mat &lt;- .z_norm(assays(ms)[[k]][gs, ])
    m &lt;- match(colnames(mat), ei$sample_id)
    cols &lt;- scales::hue_pal()(nlevels(ei$group_id))
    names(cols) &lt;- levels(ei$group_id)
    col_anno &lt;- columnAnnotation(
        df = data.frame(group_id = ei$group_id[m]),
        col = list(group_id = cols),
        gp = gpar(col = &quot;white&quot;),
        show_annotation_name = FALSE)
    Heatmap(mat, 
        column_title = k,
        name = &quot;z-normalized\nexpression&quot;,
        col = c(&quot;royalblue&quot;, &quot;cornflowerblue&quot;, &quot;white&quot;, &quot;gold&quot;, &quot;orange&quot;),
        cluster_rows = FALSE,
        cluster_columns = FALSE,
        row_names_side = &quot;left&quot;,
        row_names_gp = gpar(fontsize = 6),
        rect_gp = gpar(col = &quot;white&quot;),
        top_annotation = col_anno)
}
hms &lt;- lapply(kids, function(k) .plot_diff_hm(ms, k, tbl[[k]]$gene, ei))</code></pre>
<div id="astrocytes" class="section level5">
<h5>Astrocytes</h5>
<p><img src="figure/3-differential.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="endothelial" class="section level5">
<h5>Endothelial</h5>
<p><img src="figure/3-differential.Rmd/unnamed-chunk-4-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="excit_neuron1" class="section level5">
<h5>excit_neuron1</h5>
<p><img src="figure/3-differential.Rmd/unnamed-chunk-4-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="excit_neuron2" class="section level5">
<h5>excit_neuron2</h5>
<p><img src="figure/3-differential.Rmd/unnamed-chunk-4-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="excit_neuron3" class="section level5">
<h5>excit_neuron3</h5>
<p><img src="figure/3-differential.Rmd/unnamed-chunk-4-5.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="excit_neuron4" class="section level5">
<h5>excit_neuron4</h5>
<p><img src="figure/3-differential.Rmd/unnamed-chunk-4-6.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="excit_neuron5" class="section level5">
<h5>excit_neuron5</h5>
<p><img src="figure/3-differential.Rmd/unnamed-chunk-4-7.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="excit_neuron6" class="section level5">
<h5>excit_neuron6</h5>
<p><img src="figure/3-differential.Rmd/unnamed-chunk-4-8.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="inhib_neuron1" class="section level5">
<h5>inhib_neuron1</h5>
<p><img src="figure/3-differential.Rmd/unnamed-chunk-4-9.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="inhib_neuron2" class="section level5">
<h5>inhib_neuron2</h5>
<p><img src="figure/3-differential.Rmd/unnamed-chunk-4-10.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="inhib_neuron3" class="section level5">
<h5>inhib_neuron3</h5>
<p><img src="figure/3-differential.Rmd/unnamed-chunk-4-11.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="microglia" class="section level5">
<h5>Microglia</h5>
<p><img src="figure/3-differential.Rmd/unnamed-chunk-4-12.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="oligodendrocytes" class="section level5">
<h5>Oligodendrocytes</h5>
<p><img src="figure/3-differential.Rmd/unnamed-chunk-4-13.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="opc" class="section level5">
<h5>OPC</h5>
<p><img src="figure/3-differential.Rmd/unnamed-chunk-4-14.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="dimension-reduction" class="section level2">
<h2>Dimension reduction</h2>
<pre class="r"><code>ps &lt;- lapply(c(plotTSNE, plotUMAP), function(fun) 
    fun(sce, colour_by = &quot;cluster_id&quot;, shape_by = &quot;group_id&quot;,
        point_size = 0.8, point_alpha = 0.4) +
        scale_shape_manual(values = c(2, 3)) +
        guides(shape = guide_legend(title = &quot;group_id&quot;, ncol = 1,
            override.aes = list(size = 3, alpha = 1)),
            color = guide_legend(override.aes = list(size = 3, alpha = 1))) +
    theme_bw() + theme(aspect.ratio = 1, panel.grid.minor = element_blank()))
Warning: &#39;add_ticks&#39; is deprecated.
Use &#39;+ geom_rug(...)&#39; instead.

Warning: &#39;add_ticks&#39; is deprecated.
Use &#39;+ geom_rug(...)&#39; instead.
lgd &lt;- get_legend(ps[[1]] + theme(
    legend.position = &quot;bottom&quot;,
    legend.direction = &quot;horizontal&quot;))
ps &lt;- lapply(ps, &quot;+&quot;, theme(legend.position = &quot;none&quot;))
ps &lt;- plot_grid(plotlist = ps, nrow = 1, labels = &quot;auto&quot;) 
plot_grid(ps, lgd, ncol = 1, rel_heights = c(1, 0.2))</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/3-differential.Rmd/tsne-umap-1.png" alt="Dimension reduction plots: t-SNE (a) and UMAP (b). Cells are colored by cluster ID and shaped by group ID." width="672" />
<p class="caption">
Dimension reduction plots: t-SNE (a) and UMAP (b). Cells are colored by cluster ID and shaped by group ID.
</p>
</div>
<div id="save-sce-results" class="section level3">
<h3>Save SCE &amp; results</h3>
<pre class="r"><code>saveRDS(sce, file.path(&quot;output&quot;, &quot;MAGL-SCE_final.rds&quot;))</code></pre>
</div>
</div>
</div>
<div id="references" class="section level1">
<h1>References</h1>
<div id="refs">
<div id="ref-McCarthy2017-scater">
<p>McCarthy, Davis J, Kieran R Campbell, Quin F Wills, and Aaron T L Lun. 2017. “Scater: Pre-Processing, Quality Control, Normalization and Visualization of Single-Cell RNA-Seq Data in R.” <em>Bioinformatics</em> 33 (8):1179–86. <a href="https://doi.org/10.1093/bioinformatics/btw777" class="uri">https://doi.org/10.1093/bioinformatics/btw777</a>.</p>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 16.04.5 LTS

Matrix products: default
BLAS: /usr/local/R/R-3.5.1/lib/libRblas.so
LAPACK: /usr/local/R/R-3.5.1/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_CA.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_CA.UTF-8        LC_COLLATE=en_CA.UTF-8    
 [5] LC_MONETARY=en_CA.UTF-8    LC_MESSAGES=en_CA.UTF-8   
 [7] LC_PAPER=en_CA.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_CA.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] grid      stats4    parallel  stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] ComplexHeatmap_1.20.0       scater_1.10.1              
 [3] SingleCellExperiment_1.4.1  SummarizedExperiment_1.12.0
 [5] DelayedArray_0.8.0          BiocParallel_1.16.6        
 [7] matrixStats_0.54.0          Biobase_2.42.0             
 [9] GenomicRanges_1.34.0        GenomeInfoDb_1.18.2        
[11] IRanges_2.16.0              S4Vectors_0.20.1           
[13] BiocGenerics_0.28.0         muscat_0.99.8              
[15] magrittr_1.5                limma_3.38.3               
[17] cowplot_0.9.4               ggplot2_3.1.1              
[19] BiocStyle_2.10.0           

loaded via a namespace (and not attached):
  [1] backports_1.1.4          circlize_0.4.6          
  [3] Hmisc_4.2-0              blme_1.0-4              
  [5] workflowr_1.3.0          plyr_1.8.4              
  [7] igraph_1.2.4.1           lazyeval_0.2.2          
  [9] splines_3.5.1            listenv_0.7.0           
 [11] digest_0.6.18            foreach_1.4.4           
 [13] htmltools_0.3.6          viridis_0.5.1           
 [15] gdata_2.18.0             lmerTest_3.1-0          
 [17] checkmate_1.9.3          memoise_1.1.0           
 [19] cluster_2.0.8            doParallel_1.0.14       
 [21] globals_0.12.4           annotate_1.60.1         
 [23] prettyunits_1.0.2        colorspace_1.4-1        
 [25] blob_1.1.1               xfun_0.7                
 [27] dplyr_0.8.1              crayon_1.3.4            
 [29] RCurl_1.95-4.12          genefilter_1.64.0       
 [31] lme4_1.1-21              survival_2.44-1.1       
 [33] iterators_1.0.10         glue_1.3.1              
 [35] gtable_0.3.0             zlibbioc_1.28.0         
 [37] XVector_0.22.0           GetoptLong_0.1.7        
 [39] Rhdf5lib_1.4.3           future.apply_1.2.0      
 [41] shape_1.4.4              HDF5Array_1.10.1        
 [43] scales_1.0.0             DBI_1.0.0               
 [45] edgeR_3.24.3             Rcpp_1.0.1              
 [47] viridisLite_0.3.0        xtable_1.8-4            
 [49] progress_1.2.1           htmlTable_1.13.1        
 [51] foreign_0.8-71           bit_1.1-14              
 [53] Formula_1.2-3            htmlwidgets_1.3         
 [55] gplots_3.0.1.1           RColorBrewer_1.1-2      
 [57] acepack_1.4.1            pkgconfig_2.0.2         
 [59] XML_3.98-1.19            nnet_7.3-12             
 [61] locfit_1.5-9.1           dynamicTreeCut_1.63-1   
 [63] labeling_0.3             tidyselect_0.2.5        
 [65] rlang_0.3.4              reshape2_1.4.3          
 [67] AnnotationDbi_1.44.0     munsell_0.5.0           
 [69] tools_3.5.1              RSQLite_2.1.1           
 [71] evaluate_0.13            stringr_1.4.0           
 [73] yaml_2.2.0               knitr_1.22              
 [75] bit64_0.9-7              fs_1.2.7                
 [77] caTools_1.17.1.2         purrr_0.3.2             
 [79] future_1.12.0            nlme_3.1-139            
 [81] scran_1.10.2             pbkrtest_0.4-7          
 [83] compiler_3.5.1           rstudioapi_0.10         
 [85] beeswarm_0.2.3           variancePartition_1.12.3
 [87] tibble_2.1.1             statmod_1.4.30          
 [89] geneplotter_1.60.0       stringi_1.4.3           
 [91] highr_0.8                lattice_0.20-38         
 [93] Matrix_1.2-17            nloptr_1.2.1            
 [95] pillar_1.4.0             BiocManager_1.30.4      
 [97] GlobalOptions_0.1.0      BiocNeighbors_1.0.0     
 [99] data.table_1.12.2        bitops_1.0-6            
[101] colorRamps_2.3           R6_2.4.0                
[103] latticeExtra_0.6-28      KernSmooth_2.23-15      
[105] gridExtra_2.3            vipor_0.4.5             
[107] codetools_0.2-16         gtools_3.8.1            
[109] boot_1.3-20              MASS_7.3-51.3           
[111] assertthat_0.2.1         rhdf5_2.26.2            
[113] DESeq2_1.22.2            rprojroot_1.3-2         
[115] rjson_0.2.20             withr_2.1.2             
[117] sctransform_0.2.0        GenomeInfoDbData_1.2.0  
[119] hms_0.4.2                rpart_4.1-15            
[121] minqa_1.2.4              rmarkdown_1.12          
[123] DelayedMatrixStats_1.4.0 git2r_0.25.2            
[125] numDeriv_2016.8-1        base64enc_0.1-3         
[127] ggbeeswarm_0.6.0        </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
