<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />



<meta name="date" content="2019-07-11" />

<title>Visualization</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">LPS</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Contents</a>
</li>
<li>
  <a href="0-preprocessing.html">Preprocessing</a>
</li>
<li>
  <a href="1-clustering.html">Clustering</a>
</li>
<li>
  <a href="2-annotation.html">Annotation</a>
</li>
<li>
  <a href="3-differential.html">DS analysis</a>
</li>
<li>
  <a href="4-geneset.html">Geneset analysis</a>
</li>
<li>
  <a href="5-visualization.html">Visualization</a>
</li>
<li>
  <a href="6-more.html">Downstream</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://robinsonlabuzh.github.io">Robinson Lab</a>
</li>
<li>
  <a href="https://github.com/HelenaLC/muscat-comparison">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Visualization</h1>
<h4 class="author">Helena L. Crowell</h4>
<address class="author_afil">
Institute for Molecular Life Sciences, University of Zurich, Switzerland<br><h4 class="date">July 11, 2019</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-07-11
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 4 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 3
</p>
<p>
<strong>Knit directory:</strong> <code>MAGL/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.4.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div class="panel-group" id="workflowr-checks">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongobjectspresent"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>Environment:</strong> objects present </a>
</p>
</div>
<div id="strongEnvironmentstrongobjectspresent" class="panel-collapse collapse">
<div class="panel-body">

<p>
The global environment had objects present when the code in the R Markdown file was run. These objects can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment. Use <code>wflow_publish</code> or <code>wflow_build</code> to ensure that the code is always run in an empty environment.
</p>
<p>
The following objects were defined in the global environment when these results were created:
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
Name
</th>
<th style="text-align:left;">
Class
</th>
<th style="text-align:left;">
Size
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
data
</td>
<td style="text-align:left;">
environment
</td>
<td style="text-align:left;">
56 bytes
</td>
</tr>
<tr>
<td style="text-align:left;">
env
</td>
<td style="text-align:left;">
environment
</td>
<td style="text-align:left;">
56 bytes
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190311code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190311)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190311code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190311)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongdetected"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>Cache:</strong> detected </a>
</p>
</div>
<div id="strongCachestrongdetected" class="panel-collapse collapse">
<div class="panel-body">
The following chunks had caches available:
<ul>
<li>
diff-hm
</li>
<li>
dr
</li>
<li>
session-info-chunk-inserted-by-workflowr
</li>
<li>
unnamed-chunk-1
</li>
<li>
unnamed-chunk-2
</li>
<li>
unnamed-chunk-3
</li>
<li>
unnamed-chunk-4
</li>
<li>
violins
</li>
</ul>
<p>To ensure reproducibility of the results, delete the cache directory <code>4-visualization_cache</code> and re-run the analysis. To have workflowr automatically delete the cache directory prior to building the file, set <code>delete_cache = TRUE</code> when running <code>wflow_build()</code> or <code>wflow_publish()</code>.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomHelenaLCmuscatcomparisontree244411f3466962bb727005a8b10b41349a588742targetblank244411fa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/HelenaLC/muscat-comparison/tree/244411f3466962bb727005a8b10b41349a588742" target="_blank">244411f</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomHelenaLCmuscatcomparisontree244411f3466962bb727005a8b10b41349a588742targetblank244411fa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    ._.Renviron
    Ignored:    ._.gitignore
    Ignored:    ._README.md
    Ignored:    ._Snakefile
    Ignored:    .snakemake/
    Ignored:    MAGL/.DS_Store
    Ignored:    MAGL/.RData
    Ignored:    MAGL/.Rhistory
    Ignored:    MAGL/._.DS_Store
    Ignored:    MAGL/._.Rprofile
    Ignored:    MAGL/._.gitignore
    Ignored:    MAGL/._data
    Ignored:    MAGL/analysis/._refs.bib
    Ignored:    MAGL/analysis/0-preprocessing_cache/
    Ignored:    MAGL/analysis/1-clustering_cache/
    Ignored:    MAGL/analysis/3-differential_cache/
    Ignored:    MAGL/analysis/4-visualization_cache/
    Ignored:    MAGL/analysis/5-geneset_cache/
    Ignored:    MAGL/analysis/6-more_cache/
    Ignored:    MAGL/data/
    Ignored:    MAGL/output/
    Ignored:    data/sim_data/
    Ignored:    figures/
    Ignored:    meta/
    Ignored:    results/
    Ignored:    scripts/._apply_scdd.R
    Ignored:    scripts/._fig_vehicle_vs_lps.R
    Ignored:    scripts/._plot_perf_by_expr.R
    Ignored:    scripts/._plot_perf_by_lfc.R
    Ignored:    scripts/._plot_perf_by_ss.R
    Ignored:    scripts/._plot_runtimes.R
    Ignored:    scripts/._plot_sim_ex.R
    Ignored:    scripts/._sim_qc.R

Untracked files:
    Untracked:  .RData
    Untracked:  .RDataTmp
    Untracked:  .Rhistory
    Untracked:  scripts/fig_vehicle_vs_lps.R

Unstaged changes:
    Modified:   MAGL/.Rprofile
    Modified:   MAGL/MAGL.Rproj
    Modified:   MAGL/README.md
    Modified:   MAGL/_workflowr.yml
    Modified:   MAGL/analysis/0-preprocessing.Rmd
    Modified:   MAGL/analysis/1-clustering.Rmd
    Modified:   MAGL/analysis/3-differential.Rmd
    Modified:   MAGL/analysis/4-visualization.Rmd
    Modified:   MAGL/analysis/6-more.Rmd
    Modified:   MAGL/analysis/_site.yml
    Modified:   MAGL/analysis/index.Rmd
    Modified:   MAGL/code/README.md
    Modified:   MAGL/docs/.nojekyll
    Modified:   README.md
    Modified:   scripts/plot_lfc.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/HelenaLC/muscat-comparison/blob/08bf26003c87dc52a3395028d1117b7a0ffb75a9/MAGL/analysis/4-visualization.Rmd" target="_blank">08bf260</a>
</td>
<td>
HelenaLC
</td>
<td>
2019-06-17
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/HelenaLC/muscat-comparison/08bf26003c87dc52a3395028d1117b7a0ffb75a9/MAGL/docs/4-visualization.html" target="_blank">08bf260</a>
</td>
<td>
HelenaLC
</td>
<td>
2019-06-17
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(cowplot)
library(ComplexHeatmap)
library(dplyr)
library(ggplot2)
library(muscat)
library(RColorBrewer)
library(purrr)
library(reshape2)
library(scater)
library(SingleCellExperiment)

options(future.globals.maxSize = 2e3*1024^2)</code></pre>
<div id="load-data-results" class="section level1">
<h1>Load data &amp; results</h1>
<pre class="r"><code>sce &lt;- readRDS(file.path(&quot;output&quot;, &quot;MAGL-SCE.rds&quot;))
res &lt;- readRDS(file.path(&quot;output&quot;, &quot;MAGL-DS_res.rds&quot;))</code></pre>
</div>
<div id="results-filtering-overview" class="section level1">
<h1>Results filtering &amp; overview</h1>
<p>To get a general overview of the differential testing results, we first filter them to retain hits with <span class="math inline">\(\text{FDR}&lt;5\%\)</span> and <span class="math inline">\(~|~logFC~|~&gt;1\)</span>, and view the number &amp; percentage of differential findings by cluster. Finally, we extract the top hits (lowest adj. p-value) in each cluster.</p>
<pre class="r"><code>tbl &lt;- res$table$`LPS-WT` %&gt;% 
    # filter abs(logFC) &gt; 1 &amp; FDR &lt; 0.05
    lapply(dplyr::filter, p_adj.loc &lt; 0.05, abs(logFC) &gt; 1) %&gt;% 
    # sort by FDR
    lapply(dplyr::arrange, p_adj.loc)

# nb. &amp; % of hits by cluster
n_de &lt;- vapply(tbl, nrow, numeric(1))
cbind(n_de, perc = round(n_de / nrow(sce) * 100, 2))</code></pre>
<pre><code>                 n_de perc
Astrocytes        559 4.94
Endothelial       485 4.29
Microglia         327 2.89
OPC               167 1.48
Oligodendrocytes  214 1.89
excit_neuron1      87 0.77
excit_neuron2      26 0.23
excit_neuron3      45 0.40
excit_neuron4       0 0.00
excit_neuron5       9 0.08
excit_neuron6      17 0.15
inhib_neuron1       1 0.01
inhib_neuron2       5 0.04
inhib_neuron3       1 0.01</code></pre>
</div>
<div id="between-cluster-concordance" class="section level1">
<h1>Between-cluster concordance</h1>
<pre class="r"><code>nk &lt;- length(kids &lt;- levels(sce$cluster_id) %&gt;% set_names)
cmat &lt;- sapply(kids, function(k1) sapply(kids, function(k2)
    sum(tbl[[k1]]$gene %in% tbl[[k2]]$gene) / nrow(tbl[[k1]])))

n_de &lt;- sapply(tbl, nrow)
plot_grid(
    ggplot(data.frame(cluster_id = kids, n_de),
        aes(x = cluster_id, y = n_de)) +
        geom_bar(stat = &quot;identity&quot;, size = 0.2, width = 0.8, 
            fill = &quot;lightgrey&quot;, col = &quot;darkgrey&quot;) +
        annotate(&quot;rect&quot;, xmin = 0.5, xmax = nk + 0.5, ymin = 602, ymax = 700, fill = &quot;white&quot;) +
        annotate(&quot;text&quot;, x = seq_len(nk), y = 650, label = n_de, size = 4, fontface = &quot;bold&quot;) +
        scale_x_discrete(limits = kids) +
        scale_y_continuous(NULL, limits = c(0, 700), 
            breaks = seq(0, 600, 200), expand = c(0, 0), 
            sec.axis = dup_axis(name = &quot;Nb. of DE genes&quot;)) +
        theme_minimal() + theme(
            axis.title.x = element_blank(),
            axis.text.x = element_blank(),
            panel.grid.minor = element_blank(),
            axis.text.y.right = element_blank(),
            axis.ticks.y.right = element_blank()),
    ggplot(melt(cmat * 100), 
        aes(x = Var1, y = Var2, fill = value)) +
        geom_tile(col = &quot;white&quot;, size = 0.2) +
        geom_text(aes(label = ifelse(is.na(value), &quot;&quot;, round(value, 1))), size = 4) +
        scale_fill_distiller(&quot;Overlap (%)&quot;, palette = &quot;RdYlBu&quot;, na.value = &quot;lightgrey&quot;) +
        scale_x_discrete(limits = kids) + scale_y_discrete(limits = rev(kids)) +
        theme_minimal() + theme(
            panel.grid = element_blank(),
            axis.title = element_blank(),
            axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)),
    ncol = 1, axis = &quot;lr&quot;, align = &quot;v&quot;, rel_heights = c(1, 4))</code></pre>
<p><img src="figure/4-visualization.Rmd/unnamed-chunk-4-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="cell-level-visualization" class="section level1 tabset">
<h1>Cell-level visualization</h1>
<p>The <em><a href="https://bioconductor.org/packages/3.9/scater">scater</a></em><span class="citation">[@McCarthy2017-scater]</span> packages provides a variety of visualizations for single-cell data. Here, we use <code>plotExpression</code> to render violin plots of the top differential genes identified for each cluster. We specify <code>x = "sample_id"</code> to obtain one violin per sample, and <code>colour_by = "group_id"</code> to signify the experimental condition each sample belongs to.</p>
<pre class="r"><code># split cells by cluster
cs_by_k &lt;- split(colnames(sce), sce$cluster_id)
# get top-hits for ea. cluster
n &lt;- 8
top_gs &lt;- lapply(tbl, function(u) {
    n &lt;- ifelse(nrow(u) &lt; n, nrow(u), n)
    u$gene[seq_len(n)]
})
for (k in kids) {
    cat(&quot;## &quot;, k, &quot;\n&quot;)
    if (length(top_gs[[k]]) != 0)
        print(plotExpression(
            sce[, cs_by_k[[k]]], features = top_gs[[k]], 
            x = &quot;sample_id&quot;, colour_by = &quot;group_id&quot;) +
        guides(fill = guide_legend(override.aes = list(size = 2, alpha = 1))) +
        theme_classic() + theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)))
    cat(&quot;\n\n&quot;)
}</code></pre>
<div id="astrocytes" class="section level2">
<h2>Astrocytes</h2>
<p><img src="figure/4-visualization.Rmd/violins-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="endothelial" class="section level2">
<h2>Endothelial</h2>
<p><img src="figure/4-visualization.Rmd/violins-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="microglia" class="section level2">
<h2>Microglia</h2>
<p><img src="figure/4-visualization.Rmd/violins-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="opc" class="section level2">
<h2>OPC</h2>
<p><img src="figure/4-visualization.Rmd/violins-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="oligodendrocytes" class="section level2">
<h2>Oligodendrocytes</h2>
<p><img src="figure/4-visualization.Rmd/violins-5.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="excit_neuron1" class="section level2">
<h2>excit_neuron1</h2>
<p><img src="figure/4-visualization.Rmd/violins-6.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="excit_neuron2" class="section level2">
<h2>excit_neuron2</h2>
<p><img src="figure/4-visualization.Rmd/violins-7.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="excit_neuron3" class="section level2">
<h2>excit_neuron3</h2>
<p><img src="figure/4-visualization.Rmd/violins-8.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="excit_neuron4" class="section level2">
<h2>excit_neuron4</h2>
</div>
<div id="excit_neuron5" class="section level2">
<h2>excit_neuron5</h2>
<p><img src="figure/4-visualization.Rmd/violins-9.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="excit_neuron6" class="section level2">
<h2>excit_neuron6</h2>
<p><img src="figure/4-visualization.Rmd/violins-10.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="inhib_neuron1" class="section level2">
<h2>inhib_neuron1</h2>
<p><img src="figure/4-visualization.Rmd/violins-11.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="inhib_neuron2" class="section level2">
<h2>inhib_neuron2</h2>
<p><img src="figure/4-visualization.Rmd/violins-12.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="inhib_neuron3" class="section level2">
<h2>inhib_neuron3</h2>
<p><img src="figure/4-visualization.Rmd/violins-13.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="sample-level-visualization" class="section level1 tabset">
<h1>Sample-level visualization</h1>
<pre class="r"><code>for (k in kids) {
    cat(&quot;## &quot;, k, &quot;\n&quot;)
    hm &lt;- plotDiffHeatmap(sce, res, k = k, top_n = 20)
    tryCatch(print(hm), error = function(e) NULL)
    cat(&quot;\n\n&quot;)
}</code></pre>
<div id="astrocytes-1" class="section level2">
<h2>Astrocytes</h2>
<p><img src="figure/4-visualization.Rmd/diff-hm-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="endothelial-1" class="section level2">
<h2>Endothelial</h2>
<p><img src="figure/4-visualization.Rmd/diff-hm-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="microglia-1" class="section level2">
<h2>Microglia</h2>
<p><img src="figure/4-visualization.Rmd/diff-hm-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="opc-1" class="section level2">
<h2>OPC</h2>
<p><img src="figure/4-visualization.Rmd/diff-hm-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="oligodendrocytes-1" class="section level2">
<h2>Oligodendrocytes</h2>
<p><img src="figure/4-visualization.Rmd/diff-hm-5.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="excit_neuron1-1" class="section level2">
<h2>excit_neuron1</h2>
<p><img src="figure/4-visualization.Rmd/diff-hm-6.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="excit_neuron2-1" class="section level2">
<h2>excit_neuron2</h2>
<p><img src="figure/4-visualization.Rmd/diff-hm-7.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="excit_neuron3-1" class="section level2">
<h2>excit_neuron3</h2>
<p><img src="figure/4-visualization.Rmd/diff-hm-8.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="excit_neuron4-1" class="section level2">
<h2>excit_neuron4</h2>
</div>
<div id="excit_neuron5-1" class="section level2">
<h2>excit_neuron5</h2>
<p><img src="figure/4-visualization.Rmd/diff-hm-9.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="excit_neuron6-1" class="section level2">
<h2>excit_neuron6</h2>
<p><img src="figure/4-visualization.Rmd/diff-hm-10.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="inhib_neuron1-1" class="section level2">
<h2>inhib_neuron1</h2>
<p><img src="figure/4-visualization.Rmd/diff-hm-11.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="inhib_neuron2-1" class="section level2">
<h2>inhib_neuron2</h2>
<p><img src="figure/4-visualization.Rmd/diff-hm-12.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="inhib_neuron3-1" class="section level2">
<h2>inhib_neuron3</h2>
<p><img src="figure/4-visualization.Rmd/diff-hm-13.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="dimension-reduction" class="section level1 tabset">
<h1>Dimension reduction</h1>
<pre class="r"><code>for (dr in c(&quot;TSNE&quot;, &quot;UMAP&quot;)) {
    cat(&quot;## &quot;, dr, &quot;\n&quot;) 
    df &lt;- data.frame(reducedDim(sce, dr), colData(sce))
    labs &lt;- paste(switch(dr, TSNE = &quot;t-SNE&quot;, dr), &quot;dim.&quot;, 1:2)
    p &lt;- ggplot(df, aes_string(x = colnames(df)[1], y = colnames(df)[2])) + 
        labs(x = labs[1], y = labs[2]) +
        theme_void() + theme(aspect.ratio = 1, 
            legend.key.height = unit(2, &quot;mm&quot;), 
            legend.position = &quot;bottom&quot;)
    p1 &lt;-  p + geom_point(size = 0.1, alpha = 0.2, aes(col = cluster_id)) + 
        scale_color_manual(values = muscat:::cluster_colors) +
        guides(color = guide_legend(ncol = 3,
            override.aes = list(size = 2, alpha = 1)))
    p2 &lt;- p + geom_point(size = 0.1, alpha = 0.2, aes(col = group_id)) +
        guides(color = guide_legend(ncol = 1,
            override.aes = list(size = 2, alpha = 1)))
    ps &lt;- plot_grid(
        p1 + theme(legend.position = &quot;none&quot;),
        p2 + theme(legend.position = &quot;none&quot;))
    ls &lt;- plot_grid(get_legend(p1), get_legend(p2), rel_widths = c(4, 1))
    print(plot_grid(ps, ls, ncol = 1, rel_heights = c(3, 1)))
    cat(&quot;\n\n&quot;)
}</code></pre>
<div id="tsne" class="section level2">
<h2>TSNE</h2>
<p><img src="figure/4-visualization.Rmd/dr-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="umap" class="section level2">
<h2>UMAP</h2>
<p><img src="figure/4-visualization.Rmd/dr-2.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 16.04.5 LTS

Matrix products: default
BLAS: /usr/local/R/R-3.5.1/lib/libRblas.so
LAPACK: /usr/local/R/R-3.5.1/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_CA.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_CA.UTF-8        LC_COLLATE=en_CA.UTF-8    
 [5] LC_MONETARY=en_CA.UTF-8    LC_MESSAGES=en_CA.UTF-8   
 [7] LC_PAPER=en_CA.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_CA.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] parallel  stats4    grid      stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] SingleCellExperiment_1.4.1  SummarizedExperiment_1.12.0
 [3] DelayedArray_0.8.0          BiocParallel_1.16.6        
 [5] matrixStats_0.54.0          Biobase_2.42.0             
 [7] GenomicRanges_1.34.0        GenomeInfoDb_1.18.2        
 [9] IRanges_2.16.0              S4Vectors_0.20.1           
[11] BiocGenerics_0.28.0         reshape2_1.4.3             
[13] purrr_0.3.2                 RColorBrewer_1.1-2         
[15] dplyr_0.8.1                 ComplexHeatmap_1.20.0      
[17] cowplot_0.9.4               ggplot2_3.1.1              
[19] BiocStyle_2.10.0           

loaded via a namespace (and not attached):
 [1] circlize_0.4.6         shape_1.4.4            GetoptLong_0.1.7      
 [4] tidyselect_0.2.5       xfun_0.7               lattice_0.20-38       
 [7] colorspace_1.4-1       htmltools_0.3.6        yaml_2.2.0            
[10] rlang_0.3.4            pillar_1.4.0           glue_1.3.1            
[13] withr_2.1.2            GenomeInfoDbData_1.2.0 plyr_1.8.4            
[16] zlibbioc_1.28.0        stringr_1.4.0          munsell_0.5.0         
[19] gtable_0.3.0           workflowr_1.3.0        GlobalOptions_0.1.0   
[22] codetools_0.2-16       evaluate_0.13          labeling_0.3          
[25] knitr_1.22             Rcpp_1.0.1             scales_1.0.0          
[28] backports_1.1.4        BiocManager_1.30.4     XVector_0.22.0        
[31] fs_1.3.1               rjson_0.2.20           digest_0.6.18         
[34] stringi_1.4.3          rprojroot_1.3-2        tools_3.5.1           
[37] bitops_1.0-6           magrittr_1.5           lazyeval_0.2.2        
[40] RCurl_1.95-4.12        tibble_2.1.1           crayon_1.3.4          
[43] pkgconfig_2.0.2        Matrix_1.2-17          assertthat_0.2.1      
[46] rmarkdown_1.12         R6_2.4.0               git2r_0.25.2          
[49] compiler_3.5.1        </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
