<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2020-03-31" />

<title>Visualization</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">LPS</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Contents</a>
</li>
<li>
  <a href="0-preprocessing.html">Preprocessing</a>
</li>
<li>
  <a href="1-clustering.html">Clustering</a>
</li>
<li>
  <a href="2-annotation.html">Annotation</a>
</li>
<li>
  <a href="3-differential.html">DS analysis</a>
</li>
<li>
  <a href="4-visualization.html">Visualization</a>
</li>
<li>
  <a href="5-geneset.html">Geneset analysis</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://robinsonlabuzh.github.io">Robinson Lab</a>
</li>
<li>
  <a href="https://github.com/HelenaLC/muscat-comparison">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Visualization</h1>
<h4 class="author">Helena L. Crowell</h4>
<address class="author_afil">
Institute for Molecular Life Sciences, University of Zurich, Switzerland<br><h4 class="date">March 31, 2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-03-31
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>LPS/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.1). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190311code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190311)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190311code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190311)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongdetected"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>Cache:</strong> detected </a>
</p>
</div>
<div id="strongCachestrongdetected" class="panel-collapse collapse">
<div class="panel-body">
The following chunks had caches available:
<ul>
<li>
clusters-freqs
</li>
<li>
ds-pb
</li>
<li>
go
</li>
<li>
lfc-heatmap
</li>
<li>
lfc-heatmap-ccid3
</li>
<li>
linear-decomp
</li>
<li>
load-data
</li>
<li>
load-libs
</li>
<li>
save-lfc
</li>
<li>
session-info-chunk-inserted-by-workflowr
</li>
<li>
umaps
</li>
<li>
unnamed-chunk-1
</li>
<li>
unnamed-chunk-2
</li>
<li>
unnamed-chunk-3
</li>
<li>
unnamed-chunk-4
</li>
<li>
unnamed-chunk-5
</li>
<li>
unnamed-chunk-6
</li>
<li>
upset
</li>
<li>
violins
</li>
</ul>
<p>To ensure reproducibility of the results, delete the cache directory <code>4-visualization_cache</code> and re-run the analysis. To have workflowr automatically delete the cache directory prior to building the file, set <code>delete_cache = TRUE</code> when running <code>wflow_build()</code> or <code>wflow_publish()</code>.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomHelenaLCmuscatcomparisontreebebad71d997cb293d5a6b6f1dcb6502257ded59ctargetblankbebad71a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/HelenaLC/muscat-comparison/tree/bebad71d997cb293d5a6b6f1dcb6502257ded59c" target="_blank">bebad71</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomHelenaLCmuscatcomparisontreebebad71d997cb293d5a6b6f1dcb6502257ded59ctargetblankbebad71a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/HelenaLC/muscat-comparison/tree/bebad71d997cb293d5a6b6f1dcb6502257ded59c" target="_blank">bebad71</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .snakemake/
    Ignored:    LPS/.DS_Store
    Ignored:    LPS/analysis/.DS_Store
    Ignored:    LPS/analysis/0-preprocessing_cache/
    Ignored:    LPS/analysis/1-clustering_cache/
    Ignored:    LPS/analysis/2-annotation_cache/
    Ignored:    LPS/analysis/3-differential_cache/
    Ignored:    LPS/analysis/4-visualization_cache/
    Ignored:    LPS/analysis/5-geneset_cache/
    Ignored:    LPS/code/.DS_Store
    Ignored:    LPS/data/
    Ignored:    LPS/figures/
    Ignored:    LPS/output/.DS_Store
    Ignored:    LPS/output/DS_results_AD-gid.logcounts.rds
    Ignored:    LPS/output/DS_results_AD-gid.vstresiduals.rds
    Ignored:    LPS/output/DS_results_AD-sid.logcounts.rds
    Ignored:    LPS/output/DS_results_AD-sid.vstresiduals.rds
    Ignored:    LPS/output/DS_results_MAST.logcounts.rds
    Ignored:    LPS/output/DS_results_MM-dream.rds
    Ignored:    LPS/output/DS_results_MM-dream2.rds
    Ignored:    LPS/output/DS_results_MM-nbinom.rds
    Ignored:    LPS/output/DS_results_MM-vst.rds
    Ignored:    LPS/output/DS_results_edgeR.sum.counts.rds
    Ignored:    LPS/output/DS_results_edgeR.sum.scalecpm.rds
    Ignored:    LPS/output/DS_results_limma-trend.mean.logcounts.rds
    Ignored:    LPS/output/DS_results_limma-trend.mean.vstresiduals.rds
    Ignored:    LPS/output/DS_results_limma-voom.sum.counts.rds
    Ignored:    LPS/output/DS_results_pb.csv
    Ignored:    LPS/output/DS_results_scDD.logcounts.rds
    Ignored:    LPS/output/DS_results_scDD.vstresiduals.rds
    Ignored:    LPS/output/GSEA_data.rds
    Ignored:    LPS/output/GSEA_df.rds
    Ignored:    LPS/output/GSEA_results.csv
    Ignored:    LPS/output/SCE_annotation.rds
    Ignored:    LPS/output/SCE_preprocessing.rds
    Ignored:    LPS/output/SO_clustering.rds
    Ignored:    LPS/output/logFCs_by_cluster-sample.rds
    Ignored:    LPS/output/topGO_ccid3.csv
    Ignored:    data/
    Ignored:    figures/.DS_Store
    Ignored:    figures/kang-perf_by_cat.pdf
    Ignored:    figures/kang-perf_combined.pdf
    Ignored:    figures/magl-perf_by_cat.pdf
    Ignored:    figures/null.pdf
    Ignored:    figures/pb_mean_disp.pdf
    Ignored:    figures/perf_by_es.pdf
    Ignored:    figures/perf_by_gs.pdf
    Ignored:    figures/perf_by_ss.pdf
    Ignored:    figures/sim_pars.pdf
    Ignored:    figures/sim_vs_est_lfc.pdf
    Ignored:    figures/summary_heatmap.pdf
    Ignored:    meta/
    Ignored:    plots/
    Ignored:    results/
    Ignored:    scripts/.DS_Store
    Ignored:    scripts/.drake/config/
    Ignored:    scripts/.drake/data/
    Ignored:    scripts/.drake/drake/
    Ignored:    scripts/.drake/keys/
    Ignored:    scripts/.drake/scratch/

Untracked files:
    Untracked:  LPS/workflowr.Rproj
    Untracked:  README.html
    Untracked:  figures/figures/
    Untracked:  logs/
    Untracked:  session_info.txt
    Untracked:  setup.Rout

Unstaged changes:
    Modified:   .Renviron
    Modified:   LPS/.gitignore
    Modified:   LPS/analysis/0-preprocessing.Rmd
    Modified:   LPS/analysis/1-clustering.Rmd
    Modified:   LPS/analysis/2-annotation.Rmd
    Modified:   LPS/analysis/3-differential.Rmd
    Modified:   LPS/analysis/4-visualization.Rmd
    Modified:   LPS/analysis/5-geneset.Rmd
    Deleted:    LPS/analysis/6-more.Rmd
    Modified:   LPS/analysis/_site.yml
    Modified:   LPS/analysis/index.Rmd
    Modified:   LPS/code/utils.R
    Modified:   README.md
    Modified:   Snakefile
    Modified:   figures/fig_sim_pars.R
    Modified:   figures/fig_summary.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>LPS/analysis/4-visualization.Rmd</code>) and HTML (<code>LPS/docs/4-visualization.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/HelenaLC/muscat-comparison/blob/a38f6f0cf98346a3ade9d34d759334ff0b596321/LPS/analysis/4-visualization.Rmd" target="_blank">a38f6f0</a>
</td>
<td>
HelenaLC
</td>
<td>
2019-08-13
</td>
<td>
bioc3.10 rerun
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/HelenaLC/muscat-comparison/a38f6f0cf98346a3ade9d34d759334ff0b596321/LPS/docs/4-visualization.html" target="_blank">a38f6f0</a>
</td>
<td>
HelenaLC
</td>
<td>
2019-08-13
</td>
<td>
bioc3.10 rerun
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/HelenaLC/muscat-comparison/blob/7189a8844c36c1430a3d2ca6709e7710b69b2997/LPS/analysis/4-visualization.Rmd" target="_blank">7189a88</a>
</td>
<td>
HelenaLC
</td>
<td>
2019-08-13
</td>
<td>
bioc3.10 rerun
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/HelenaLC/muscat-comparison/7189a8844c36c1430a3d2ca6709e7710b69b2997/LPS/docs/4-visualization.html" target="_blank">7189a88</a>
</td>
<td>
HelenaLC
</td>
<td>
2019-08-13
</td>
<td>
bioc3.10 rerun
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level3">
<h3>Load packages</h3>
<pre class="r"><code>library(data.table)
library(dplyr)
library(edgeR)
library(M3C)
library(magrittr)
library(muscat)
library(purrr)
library(org.Mm.eg.db)
library(scater)
library(topGO)
library(UpSetR)</code></pre>
</div>
<div id="load-data" class="section level3">
<h3>Load data</h3>
<pre class="r"><code># load data &amp; source utilities
sce &lt;- readRDS(file.path(&quot;output&quot;, &quot;SCE_annotation.rds&quot;))
source(file.path(&quot;code&quot;, &quot;utils.R&quot;))</code></pre>
</div>
<div id="dimensionality-reduction" class="section level1 tabset">
<h1>Dimensionality reduction</h1>
<pre class="r"><code># for ea. cluster, sample equal group sizes
cs &lt;- split(setDT(cd_df), 
    by = c(&quot;cluster_id&quot;, &quot;group_id&quot;), 
    flatten = FALSE, sorted = TRUE) %&gt;% 
    map_depth(2, &quot;cell_id&quot;) %&gt;% 
    map_depth(1, function(u) 
        lapply(u, sample, min(sapply(u, length))))
cs &lt;- unname(unlist(cs))
df &lt;- cd_df[sample(cs, length(cs)), ] # randomize rows

# plot UMAP colored by cluster, sample &amp; group ID
for (id in ids) {
    cat(&quot;## &quot;, id, &quot;\n&quot;)
    print(.plot_dr(df, id, get(paste0(id, &quot;_pal&quot;))))
    cat(&quot;\n\n&quot;)
}</code></pre>
<div id="cluster_id" class="section level2">
<h2>cluster_id</h2>
<p><img src="figure/4-visualization.Rmd/umaps-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="group_id" class="section level2">
<h2>group_id</h2>
<p><img src="figure/4-visualization.Rmd/umaps-2.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="sample_id" class="section level2">
<h2>sample_id</h2>
<p><img src="figure/4-visualization.Rmd/umaps-3.png" width="768" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="relative-cluster-abundances" class="section level1">
<h1>Relative cluster abundances</h1>
<pre class="r"><code>thm &lt;- function(pal, p) p + 
    facet_wrap(&quot;group_id&quot;, ncol = 1, scales = &quot;free_y&quot;) +
    geom_bar(stat = &quot;identity&quot;, col = &quot;white&quot;,  width = 1, size = 0.2) + 
    scale_fill_manual(values = pal) + scale_x_discrete(expand = c(0, 0)) +
    .prettify(&quot;bw&quot;) + coord_flip() + theme(aspect.ratio = NULL, 
        panel.grid = element_blank(), panel.spacing = unit(1, &quot;mm&quot;), 
        strip.text = element_blank(),  strip.background = element_blank()) 

# cluster frequencies by sample
fqs &lt;- prop.table(table(sce$cluster_id, sce$sample_id), 2)
df &lt;- set_colnames(reshape2::melt(fqs), c(&quot;cluster_id&quot;, &quot;sample_id&quot;, &quot;frequency&quot;))
df$group_id &lt;- sce$group_id[match(df$sample_id, sce$sample_id)]

(p1 &lt;- thm(cluster_id_pal, ggplot(df,
    aes(x = sample_id, y = frequency, fill = cluster_id)) +
    scale_y_continuous(breaks = seq(0, 1, 0.2), expand = c(0,0))))</code></pre>
<p><img src="figure/4-visualization.Rmd/clusters-freqs-1.png" width="576" style="display: block; margin: auto;" /></p>
<pre class="r"><code># excluding neuronal subpopulations
sub &lt;- sce[, !sce$cluster_id %in% c(&quot;Excit. Neuron&quot;, &quot;Inhib. Neuron&quot;)]
fqs &lt;- prop.table(table(sub$cluster_id, sub$sample_id), 2)
df &lt;- set_colnames(reshape2::melt(fqs), c(&quot;cluster_id&quot;, &quot;sample_id&quot;, &quot;frequency&quot;))
df$group_id &lt;- sub$group_id[match(df$sample_id, sub$sample_id)]

(p2 &lt;- thm(cluster_id_pal, ggplot(df, 
    aes(x = sample_id, y = frequency, fill = cluster_id)) +
    scale_y_continuous(breaks = seq(0, 1, 0.2), expand = c(0,0))))</code></pre>
<p><img src="figure/4-visualization.Rmd/clusters-freqs-2.png" width="576" style="display: block; margin: auto;" /></p>
<pre class="r"><code># cell counts by sample
df &lt;- set_colnames(reshape2::melt(table(sce$sample_id)), c(&quot;sample_id&quot;, &quot;n_cells&quot;))
df$group_id &lt;- sce$group_id[match(df$sample_id, sce$sample_id)]

(p3 &lt;- thm(group_id_pal, ggplot(df, 
    aes(x = sample_id, y = n_cells, fill = group_id)) +
    scale_y_continuous(&quot;nb. of cells&quot;, limits = c(0, 5e3), 
        breaks = seq(0, 5e3, 2500), expand = c(0,0))))</code></pre>
<p><img src="figure/4-visualization.Rmd/clusters-freqs-3.png" width="576" style="display: block; margin: auto;" /></p>
</div>
<div id="filter-format-results" class="section level1">
<h1>Filter &amp; format results</h1>
<p>To get a general overview of the differential testing results, we first filter them to retain hits with <span class="math inline">\(\text{FDR}&lt;5\%\)</span> and <span class="math inline">\(~|~logFC~|~&gt;1\)</span>, and view the number &amp; percentage of differential findings by cluster. Finally, we extract the top hits (lowest adj. p-value) in each cluster.</p>
<pre class="r"><code># compute pseudobulks (sum of counts) by cluster-sample &amp; run DS analysis
pbs &lt;- aggregateData(sce[, sce$cluster_id != &quot;CPE cells&quot;])
res &lt;- pbDS(pbs, method = &quot;edgeR&quot;, filter = &quot;genes&quot;, verbose = FALSE)$table$LPS

# filter abs(logFC) &gt; 1 &amp; FDR &lt; 0.05, 
# sort by FDR &amp; extract DS genes for ea. cluster
res_fil &lt;- .get_ds_gs(res, fdr = 0.05, lfc = 1)
ds_gs_by_k &lt;- map(res_fil, &quot;gene&quot;)

# nb. &amp; % of hits by cluster
n_de &lt;- vapply(ds_gs_by_k, length, numeric(1))
cbind(n_de, perc = round(n_de / nrow(sce) * 100, 2))</code></pre>
<pre><code>                 n_de perc
Astrocytes        442 3.99
Endothelial       442 3.99
Microglia          93 0.84
Oligodendrocytes   53 0.48
OPC                98 0.88
Excit. Neuron       6 0.05
Inhib. Neuron       2 0.02</code></pre>
</div>
<div id="between-cluster-concordance" class="section level1">
<h1>Between-cluster concordance</h1>
<pre class="r"><code>(upset &lt;- upset(fromList(ds_gs_by_k), nintersects = 100,  
    sets = names(ds_gs_by_k), sets.bar.color = cluster_id_pal[names(ds_gs_by_k)],
    mb.ratio = c(0.6, 0.4), text.scale = 0.8, point.size = 2, line.size = 0.4))</code></pre>
<p><img src="figure/4-visualization.Rmd/upset-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
<div id="heatmap-of-cluster-wise-logfcs" class="section level1">
<h1>Heatmap of cluster-wise logFCs</h1>
<pre class="r"><code># construct SCE containing logFCs for ea. cluster
ref &lt;- pbs$group_id == &quot;Vehicle&quot;
lfc &lt;- lapply(assays(pbs), function(u) {
    y &lt;- DGEList(u)
    y &lt;- calcNormFactors(y)
    lcpm &lt;- log1p(cpm.DGEList(y))
    lfc &lt;- lcpm - rowMeans(lcpm[, ref])
    SingleCellExperiment(lfc, 
        colData = colData(pbs),
        rowData = rowData(sce)) 
}) %&gt;% do.call(what = cbind)
lfc$cluster_id &lt;- factor(
    rep(assayNames(pbs), each = ns), 
    levels = assayNames(pbs))

# consensus clustering of DS genes
ds_gs &lt;- unique(unlist(ds_gs_by_k))
cc &lt;- M3C(t(assay(lfc[ds_gs, ])), method = 2, 
    seed = NULL, removeplots = TRUE)
cc_ids &lt;- cc$realdataresults[[3]]$assignments
cc_cols &lt;- brewer.pal(length(unique(cc_ids)), &quot;Set2&quot;)
names(cc_cols) &lt;- unique(cc_ids)

# column &amp; row annoations 
col_df &lt;- data.frame(
    cluster_id = rep(assayNames(pbs), each = ns), 
    group_id = rep(pbs$group_id, length(assays(pbs))))
col_cols &lt;- sapply(ids, function(id) get(paste0(id, &quot;_pal&quot;)))
col_anno &lt;- HeatmapAnnotation(
    df = col_df, col = col_cols,
    show_annotation_name = FALSE,
    annotation_legend_param = .hm_lgd_aes)

row_anno &lt;- rowAnnotation(
    show_annotation_name = FALSE,
    df = data.frame(consensus_id = cc_ids),
    col = list(consensus_id = cc_cols),
    simple_anno_size = unit(2, &quot;mm&quot;),
    annotation_legend_param = .hm_lgd_aes)

qs &lt;- quantile(assay(lfc[ds_gs, ]), c(0.01, 0.99))
brks &lt;- .get_brks(qs, 0.5)
cols &lt;- colorRamp2(brks, .hm_cols)
(hm &lt;- .plot_hm(lfc[ds_gs, ], cols, brks, 
    col_anno, row_anno, cc_ids, row_title = NULL, 
    cluster_row_slices = FALSE, cluster_column_slices = FALSE, 
    use_raster = TRUE, raster_device = &quot;CairoPNG&quot;))</code></pre>
<p><img src="figure/4-visualization.Rmd/lfc-heatmap-1.png" width="1344" style="display: block; margin: auto;" /></p>
<div id="consensus-id-3-only" class="section level2">
<h2>Consensus ID 3 only</h2>
<pre class="r"><code># subset genes w/ consensus clustering ID 3
# &amp; more stringent filtering for 0.001% FDR
gs &lt;- intersect(
    names(which(cc_ids == 3)),
    .get_ds_gs(res, fdr = 1e-4, lfc = 1, simplify = TRUE))

qs &lt;- quantile(assay(lfc[gs, ]), c(0.01, 0.99))
brks &lt;- .get_brks(qs, 0.5)
cols &lt;- colorRamp2(brks, .hm_cols)

tmp &lt;- lfc[gs, ]
rownames(tmp) &lt;- gsub(&quot;.*[.]{1}&quot;, &quot;&quot;, rownames(tmp))
(hm &lt;- .plot_hm(tmp, cols, brks, 
    col_anno, row_title = NULL, row_nms = TRUE,
    cluster_row_slices = FALSE, cluster_column_slices = FALSE, 
    row_names_gp = gpar(fontsize = 4),
    column_title_gp = gpar(fontsize = 8),
    use_raster = TRUE, raster_device = &quot;CairoPNG&quot;))</code></pre>
<p><img src="figure/4-visualization.Rmd/lfc-heatmap-ccid3-1.png" width="1344" style="display: block; margin: auto;" /></p>
<pre><code>Loading required namespace: Cairo</code></pre>
</div>
</div>
<div id="cell-level-linear-decomposition" class="section level1">
<h1>Cell-level linear decomposition</h1>
<pre class="r"><code>cs_by_k &lt;- split(colnames(sce), sce$cluster_id)[assayNames(pbs)]
ld &lt;- lapply(assayNames(pbs), function(k) {
    # subset SCE to include only 
    # DS genes &amp; cells in cluster k
    gs &lt;- ds_gs_by_k[[k]]
    cs &lt;- cs_by_k[[k]]
    if (length(gs) == 0 | length(cs) == 0) 
        return(NULL)
    es &lt;- logcounts(sce[gs, cs])
    # decompose values as a linear combination of the group&#39;s logFCs
    ld &lt;- t(es) %*% pull(filter(res[[k]], gene %in% gs), &quot;logFC&quot;)
    # assure values are comparable across clusters
    ld[, 1] / max(ld)
})
rmv &lt;- vapply(ld, is.null, logical(1))
ld &lt;- set_names(unlist(ld[!rmv]), unlist(cs_by_k[!rmv]))
cd_df[names(ld), &quot;ld&quot;] &lt;- ld

# violins by cluster-sample
(p &lt;- ggplot(filter(cd_df, !is.na(ld)), 
    aes(x = sample_id, y = ld, fill = group_id)) +
    facet_wrap(&quot;cluster_id&quot;, nrow = 1) +
    geom_hline(yintercept = 0, size = 0.1, lty = 2) +
    scale_fill_manual(values = group_id_pal) + 
    geom_violin(size = 0.1, width = 1) + 
    geom_boxplot(size = 0.1, width = 0.2, show.legend = FALSE,
        outlier.shape = 16, outlier.size = 0.1) +
    scale_y_continuous(limits = c(-0.6, 1.1), breaks = seq(-0.5,1,0.5), expand = c(0,0)) +
    ylab(&quot;effect coefficient&quot;) + .prettify(&quot;bw&quot;) + theme(
        aspect.ratio = 1, panel.spacing = unit(2, &quot;mm&quot;),
        axis.text.x = element_text(angle = 30, hjust = 1, vjust = 1)))</code></pre>
<p><img src="figure/4-visualization.Rmd/linear-decomp-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>Scale for &#39;y&#39; is already present. Adding another scale for &#39;y&#39;, which will replace the existing
scale.</code></pre>
</div>
<div id="go-term-enrichment-analysis" class="section level1">
<h1>GO term enrichment analysis</h1>
<pre class="r"><code>go &lt;- lapply(split(names(cc_ids), cc_ids), function(gs) {
    ss1 &lt;- strsplit(gs, &quot;.&quot;, fixed = TRUE)
    ss2 &lt;- strsplit(rownames(sce), &quot;.&quot;, fixed = TRUE)
    fs1 &lt;- unique(sapply(ss1, .subset, 2))
    fs2 &lt;- sapply(ss2, .subset, 2)
    l &lt;- as.numeric(fs2 %in% fs1)
    names(l) &lt;- fs2
    go &lt;- new(&quot;topGOdata&quot;, 
        nodeSize = 20,
        ontology = &quot;BP&quot;,
        allGenes = l,
        geneSel = function(u) u == 1,
        annot = annFUN.org,
        mapping = &quot;org.Mm.eg.db&quot;,
        ID = &quot;symbol&quot;)
    res &lt;- runTest(go, algorithm = &quot;classic&quot;, statistic = &quot;fisher&quot;)
    GenTable(go, classicFisher = res, orderBy = &quot;classicFisher&quot;, topNodes = 100)
})
# view top 10 GO terms per consensus ID
lapply(go, head, 10)</code></pre>
<pre><code>$`1`
        GO.ID                               Term Annotated Significant Expected classicFisher
1  GO:0048731                 system development      2731         119    77.20       3.7e-08
2  GO:0015711            organic anion transport       224          23     6.33       8.1e-08
3  GO:0048856   anatomical structure development      3276         133    92.61       2.7e-07
4  GO:0006820                    anion transport       298          26     8.42       3.0e-07
5  GO:0007275 multicellular organism development      3034         125    85.77       3.9e-07
6  GO:0046942          carboxylic acid transport       164          18     4.64       8.2e-07
7  GO:0015849             organic acid transport       165          18     4.66       9.0e-07
8  GO:0032502              developmental process      3478         136    98.32       1.8e-06
9  GO:0032501   multicellular organismal process      3774         143   106.69       5.2e-06
10 GO:0007519 skeletal muscle tissue development       100          12     2.83       2.4e-05

$`2`
        GO.ID                     Term Annotated Significant Expected classicFisher
1  GO:0040011               locomotion      1113          96    44.61       8.5e-14
2  GO:0002376    immune system process      1161          98    46.53       1.7e-13
3  GO:0016477           cell migration       904          83    36.23       2.2e-13
4  GO:0050896     response to stimulus      4253         241   170.46       2.9e-13
5  GO:0048870            cell motility       972          84    38.96       4.3e-12
6  GO:0051674     localization of cell       972          84    38.96       4.3e-12
7  GO:0008283       cell proliferation      1102          89    44.17       3.5e-11
8  GO:0007155            cell adhesion       725          67    29.06       5.4e-11
9  GO:0040012 regulation of locomotion       644          62    25.81       5.9e-11
10 GO:0022610      biological adhesion       730          67    29.26       7.4e-11

$`3`
        GO.ID                                 Term Annotated Significant Expected classicFisher
1  GO:0006952                     defense response       657          89    11.08       &lt; 1e-30
2  GO:0006955                      immune response       597          79    10.07       &lt; 1e-30
3  GO:0051707           response to other organism       386          67     6.51       &lt; 1e-30
4  GO:0043207 response to external biotic stimulus       387          67     6.52       &lt; 1e-30
5  GO:0009607          response to biotic stimulus       405          67     6.83       &lt; 1e-30
6  GO:0002376                immune system process      1161          96    19.57       &lt; 1e-30
7  GO:0034097                 response to cytokine       464          66     7.82       &lt; 1e-30
8  GO:0009605        response to external stimulus      1136          92    19.15       &lt; 1e-30
9  GO:0045087               innate immune response       308          56     5.19       &lt; 1e-30
10 GO:0006950                   response to stress      1928         113    32.51       &lt; 1e-30</code></pre>
</div>
<div id="cell-level-visualization" class="section level1 tabset">
<h1>Cell-level visualization</h1>
<p>The <em><a href="https://bioconductor.org/packages/3.10/scater">scater</a></em> packages provides a variety of visualizations for single-cell data. Here, we use <code>plotExpression</code> to render violin plots of the top differential genes identified for each cluster. We specify <code>x = "sample_id"</code> to obtain one violin per sample, and <code>colour_by = "group_id"</code> to signify the experimental condition each sample belongs to.</p>
<pre class="r"><code># split cells by cluster
cs_by_k &lt;- split(colnames(sce), sce$cluster_id)
# get top hits for ea. cluster
n &lt;- 8
top_gs &lt;- lapply(res_fil, function(u) {
    n &lt;- ifelse(nrow(u) &lt; n, nrow(u), n)
    arrange(u, p_adj.loc) %&gt;% 
        dplyr::slice(seq_len(n)) %&gt;% 
        pull(&quot;gene&quot;)
})
for (k in assayNames(pbs)) {
    cat(&quot;## &quot;, k, &quot;\n&quot;)
    if (length(top_gs[[k]]) != 0)
        print(plotExpression(
            sce[, cs_by_k[[k]]], features = top_gs[[k]], 
            x = &quot;sample_id&quot;, colour_by = &quot;group_id&quot;) +
        guides(fill = guide_legend(override.aes = list(size = 2, alpha = 1))) +
        theme_classic() + theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)))
    cat(&quot;\n\n&quot;)
}</code></pre>
<div id="astrocytes" class="section level2">
<h2>Astrocytes</h2>
<p><img src="figure/4-visualization.Rmd/violins-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-violins-1">
Past versions of violins-1.png
</button>
</p>
<div id="fig-violins-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/HelenaLC/muscat-comparison/blob/a38f6f0cf98346a3ade9d34d759334ff0b596321/LPS/docs/figure/4-visualization.Rmd/violins-1.png" target="_blank">a38f6f0</a>
</td>
<td>
HelenaLC
</td>
<td>
2019-08-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/HelenaLC/muscat-comparison/blob/7189a8844c36c1430a3d2ca6709e7710b69b2997/LPS/docs/figure/4-visualization.Rmd/violins-1.png" target="_blank">7189a88</a>
</td>
<td>
HelenaLC
</td>
<td>
2019-08-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="endothelial" class="section level2">
<h2>Endothelial</h2>
<p><img src="figure/4-visualization.Rmd/violins-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-violins-2">
Past versions of violins-2.png
</button>
</p>
<div id="fig-violins-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/HelenaLC/muscat-comparison/blob/a38f6f0cf98346a3ade9d34d759334ff0b596321/LPS/docs/figure/4-visualization.Rmd/violins-2.png" target="_blank">a38f6f0</a>
</td>
<td>
HelenaLC
</td>
<td>
2019-08-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/HelenaLC/muscat-comparison/blob/7189a8844c36c1430a3d2ca6709e7710b69b2997/LPS/docs/figure/4-visualization.Rmd/violins-2.png" target="_blank">7189a88</a>
</td>
<td>
HelenaLC
</td>
<td>
2019-08-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="microglia" class="section level2">
<h2>Microglia</h2>
<p><img src="figure/4-visualization.Rmd/violins-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-violins-3">
Past versions of violins-3.png
</button>
</p>
<div id="fig-violins-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/HelenaLC/muscat-comparison/blob/a38f6f0cf98346a3ade9d34d759334ff0b596321/LPS/docs/figure/4-visualization.Rmd/violins-3.png" target="_blank">a38f6f0</a>
</td>
<td>
HelenaLC
</td>
<td>
2019-08-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/HelenaLC/muscat-comparison/blob/7189a8844c36c1430a3d2ca6709e7710b69b2997/LPS/docs/figure/4-visualization.Rmd/violins-3.png" target="_blank">7189a88</a>
</td>
<td>
HelenaLC
</td>
<td>
2019-08-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="oligodendrocytes" class="section level2">
<h2>Oligodendrocytes</h2>
<p><img src="figure/4-visualization.Rmd/violins-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-violins-4">
Past versions of violins-4.png
</button>
</p>
<div id="fig-violins-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/HelenaLC/muscat-comparison/blob/a38f6f0cf98346a3ade9d34d759334ff0b596321/LPS/docs/figure/4-visualization.Rmd/violins-4.png" target="_blank">a38f6f0</a>
</td>
<td>
HelenaLC
</td>
<td>
2019-08-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/HelenaLC/muscat-comparison/blob/7189a8844c36c1430a3d2ca6709e7710b69b2997/LPS/docs/figure/4-visualization.Rmd/violins-4.png" target="_blank">7189a88</a>
</td>
<td>
HelenaLC
</td>
<td>
2019-08-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="opc" class="section level2">
<h2>OPC</h2>
<p><img src="figure/4-visualization.Rmd/violins-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-violins-5">
Past versions of violins-5.png
</button>
</p>
<div id="fig-violins-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/HelenaLC/muscat-comparison/blob/a38f6f0cf98346a3ade9d34d759334ff0b596321/LPS/docs/figure/4-visualization.Rmd/violins-5.png" target="_blank">a38f6f0</a>
</td>
<td>
HelenaLC
</td>
<td>
2019-08-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/HelenaLC/muscat-comparison/blob/7189a8844c36c1430a3d2ca6709e7710b69b2997/LPS/docs/figure/4-visualization.Rmd/violins-5.png" target="_blank">7189a88</a>
</td>
<td>
HelenaLC
</td>
<td>
2019-08-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="excit.-neuron" class="section level2">
<h2>Excit. Neuron</h2>
<p><img src="figure/4-visualization.Rmd/violins-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-violins-6">
Past versions of violins-6.png
</button>
</p>
<div id="fig-violins-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/HelenaLC/muscat-comparison/blob/a38f6f0cf98346a3ade9d34d759334ff0b596321/LPS/docs/figure/4-visualization.Rmd/violins-6.png" target="_blank">a38f6f0</a>
</td>
<td>
HelenaLC
</td>
<td>
2019-08-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/HelenaLC/muscat-comparison/blob/7189a8844c36c1430a3d2ca6709e7710b69b2997/LPS/docs/figure/4-visualization.Rmd/violins-6.png" target="_blank">7189a88</a>
</td>
<td>
HelenaLC
</td>
<td>
2019-08-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="inhib.-neuron" class="section level2">
<h2>Inhib. Neuron</h2>
<p><img src="figure/4-visualization.Rmd/violins-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-violins-7">
Past versions of violins-7.png
</button>
</p>
<div id="fig-violins-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/HelenaLC/muscat-comparison/blob/a38f6f0cf98346a3ade9d34d759334ff0b596321/LPS/docs/figure/4-visualization.Rmd/violins-7.png" target="_blank">a38f6f0</a>
</td>
<td>
HelenaLC
</td>
<td>
2019-08-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/HelenaLC/muscat-comparison/blob/7189a8844c36c1430a3d2ca6709e7710b69b2997/LPS/docs/figure/4-visualization.Rmd/violins-7.png" target="_blank">7189a88</a>
</td>
<td>
HelenaLC
</td>
<td>
2019-08-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="write-logfcs-to-.rds-top-100-gos-to-.csv" class="section level1">
<h1>Write logFCs to .rds &amp; top 100 GOs to .csv</h1>
<pre class="r"><code>write.csv(head(go$`3`, 100), file.path(&quot;output&quot;, &quot;topGO_ccid3.csv&quot;))
saveRDS(lfc, file.path(&quot;output&quot;, &quot;logFCs_by_cluster-sample.rds&quot;))</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.2 (2019-12-12)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Catalina 10.15.3

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
 [1] grid      stats4    parallel  stats     graphics  grDevices utils     datasets  methods  
[10] base     

other attached packages:
 [1] purrr_0.3.3                 RColorBrewer_1.1-2          ggrastr_0.1.7              
 [4] cowplot_1.0.0               ComplexHeatmap_2.2.0        circlize_0.4.8             
 [7] UpSetR_1.4.0                topGO_2.38.1                SparseM_1.78               
[10] GO.db_3.10.0                graph_1.64.0                org.Mm.eg.db_3.10.0        
[13] AnnotationDbi_1.48.0        muscat_1.2.0                scater_1.14.6              
[16] ggplot2_3.2.1               SingleCellExperiment_1.8.0  SummarizedExperiment_1.16.1
[19] DelayedArray_0.12.2         BiocParallel_1.20.1         matrixStats_0.55.0         
[22] GenomicRanges_1.38.0        GenomeInfoDb_1.22.0         IRanges_2.20.2             
[25] S4Vectors_0.24.3            magrittr_1.5                M3C_1.8.0                  
[28] Biobase_2.46.0              BiocGenerics_0.32.0         edgeR_3.28.0               
[31] limma_3.42.2                dplyr_0.8.4                 data.table_1.12.8          

loaded via a namespace (and not attached):
  [1] tidyselect_1.0.0         lme4_1.1-21              RSQLite_2.2.0           
  [4] htmlwidgets_1.5.1        Rtsne_0.15               munsell_0.5.0           
  [7] codetools_0.2-16         future_1.16.0            withr_2.1.2             
 [10] colorspace_1.4-1         knitr_1.28               rstudioapi_0.11         
 [13] listenv_0.8.0            NMF_0.22.0               labeling_0.3            
 [16] git2r_0.26.1             sigclust_1.1.0           GenomeInfoDbData_1.2.2  
 [19] farver_2.0.3             bit64_0.9-7              glmmTMB_1.0.0           
 [22] rprojroot_1.3-2          vctrs_0.2.2              xfun_0.12               
 [25] R6_2.4.1                 doParallel_1.0.15        ggbeeswarm_0.6.0        
 [28] clue_0.3-57              rsvd_1.0.3               locfit_1.5-9.1          
 [31] bitops_1.0-6             assertthat_0.2.1         promises_1.1.0          
 [34] scales_1.1.0             nnet_7.3-12              beeswarm_0.2.3          
 [37] gtable_0.3.0             globals_0.12.5           workflowr_1.6.0         
 [40] rlang_0.4.4              genefilter_1.68.0        GlobalOptions_0.1.1     
 [43] splines_3.6.2            TMB_1.7.16               lazyeval_0.2.2          
 [46] acepack_1.4.1            checkmate_2.0.0          BiocManager_1.30.10     
 [49] yaml_2.2.1               reshape2_1.4.3           backports_1.1.5         
 [52] httpuv_1.5.2             Hmisc_4.3-1              tools_3.6.2             
 [55] gridBase_0.4-7           gplots_3.0.1.2           Rcpp_1.0.3              
 [58] plyr_1.8.5               base64enc_0.1-3          progress_1.2.2          
 [61] zlibbioc_1.32.0          RCurl_1.98-1.1           prettyunits_1.1.1       
 [64] rpart_4.1-15             GetoptLong_0.1.8         viridis_0.5.1           
 [67] cluster_2.1.0            colorRamps_2.3           fs_1.3.1                
 [70] variancePartition_1.16.1 lmerTest_3.1-1           matrixcalc_1.0-3        
 [73] whisker_0.4              hms_0.5.3                evaluate_0.14           
 [76] xtable_1.8-4             pbkrtest_0.4-7           XML_3.99-0.3            
 [79] jpeg_0.1-8.1             gridExtra_2.3            shape_1.4.4             
 [82] compiler_3.6.2           tibble_2.1.3             KernSmooth_2.23-16      
 [85] crayon_1.3.4             minqa_1.2.4              htmltools_0.4.0         
 [88] corpcor_1.6.9            later_1.0.0              Formula_1.2-3           
 [91] snow_0.4-3               geneplotter_1.64.0       DBI_1.1.0               
 [94] MASS_7.3-51.5            boot_1.3-24              BiocStyle_2.14.4        
 [97] Matrix_1.2-18            gdata_2.18.0             pkgconfig_2.0.3         
[100] registry_0.5-1           numDeriv_2016.8-1.1      foreign_0.8-75          
[103] foreach_1.4.8            annotate_1.64.0          vipor_0.4.5             
[106] rngtools_1.5             blme_1.0-4               pkgmaker_0.31           
[109] XVector_0.26.0           bibtex_0.4.2.2           stringr_1.4.0           
[112] digest_0.6.24            sctransform_0.2.1        rmarkdown_2.1           
[115] htmlTable_1.13.3         dendextend_1.13.3        DelayedMatrixStats_1.8.0
[118] gtools_3.8.1             rjson_0.2.20             nloptr_1.2.1            
[121] lifecycle_0.1.0          nlme_3.1-144             BiocNeighbors_1.4.1     
[124] viridisLite_0.3.0        pillar_1.4.3             lattice_0.20-40         
[127] survival_3.1-8           glue_1.3.1               png_0.1-7               
[130] iterators_1.0.12         bit_1.1-15.2             stringi_1.4.6           
[133] blob_1.2.1               DESeq2_1.26.0            BiocSingular_1.2.2      
[136] doSNOW_1.0.18            latticeExtra_0.6-29      caTools_1.18.0          
[139] memoise_1.1.0            irlba_2.3.3              future.apply_1.4.0      </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
