---
title: "DS analysis"
author: 
- name: Helena L. Crowell
  affiliation: 
  - &IMLS Institute for Molecular Life Sciences, University of Zurich, Switzerland
  - &SIB Swiss Institute of Bioinformatics (SIB), University of Zurich, Switzerland
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output:
  html_document:
    toc: true
    code_folding: hide
---

```{r include = FALSE}
library(BiocStyle)
knitr::opts_chunk$set(autodep = TRUE, cache = TRUE)
options(width = 100)
```

### load packages
```{r warning = FALSE, message = FALSE}
library(ComplexHeatmap)
library(cowplot)
library(limma)
library(magrittr)
library(muscat)
library(scater)
library(SingleCellExperiment)
```

# Data preparation

## Load data & prep. for `muscat`

```{r prep-data}
# load data
sce0 <- readRDS(file.path("output", "MAGL-SCE_tmp.rds"))

# concat. sample & group ID
sce0$id <- factor(paste(sce0$sample_id, sce0$group_id, sep = "_"))
# make WT reference group
sce0$group_id <- factor(sce0$group_id, levels = c("WT", "LPS"))
# reorder sample levels
o <- sapply(levels(sce0$group_id), grep, levels(sce0$id))
sce0$id <- factor(sce0$id, levels = levels(sce0$id)[o])

# prep. SCE for `muscat`
sce <- prepSCE(sce0,
    group_id = "group_id",
    sample_id = "id",
    cluster_id = "integrated_snn_res.0.2",
    drop = FALSE)

(ei <- metadata(sce)$experiment_info)
```

## Cluster annotation

In consideration of the visualizations provided in the [Annotation](docs/2-annotation.html) tab and additional exploration with `r Biocpkg("iSEE")`, we arrive at the following cluster annotations: 

```{r cluster-anno, warning = FALSE}
anno <- list(
    "excit_neuron1" = c(0, 1, 4, 5, 16),
    "excit_neuron2" = 7,
    "excit_neuron3" = c(2, 20),
    "excit_neuron4" = 9,
    "excit_neuron5" = 14,
    "excit_neuron6" = 8,
    "inhib_neuron1" = c(10, 15),
    "inhib_neuron2" = 11,
    "inhib_neuron3" = 12,
    "Astrocytes" = 3,
    "OPC" = 13,
    "Oligodendrocytes" = 6,
    "Endothelial" = 18,
    "Microglia" = 19)
# unassigned: 17, 21, 22, 23

m <- match(sce$cluster_id, unlist(anno))
ns <- vapply(anno, length, numeric(1))
lab <- rep.int(names(anno), ns)
sce$cluster_id <- lab[m]

# remove unassigned cells
sce <- sce[, !is.na(sce$cluster_id)]
sce$cluster_id <- factor(sce$cluster_id)

# cluster-sample cell-counts
table(sce$cluster_id, sce$sample_id)

# normalize for visualization
sce <- normalize(sce)
```

# Aggregation of single-cell to pseudo-bulk data

```{r agg}
system.time(pb <- aggregateData(sce))
# SCE of dim. #(genes) x #(samples)
pb 
# one assay per cluster
assayNames(pb) 
# view 1st cluster
head(assay(pb))
```

## Pseudobulk-level MDS plot

Prior to conducting any formal testing, we can compute a multi-dimensional scaling (MDS) plot of aggregated signal to explore overall sample similarities. Ideally, such a represenation of the data should separate both clusters and groups from one another. Vice versa, samples from the same cluster/group should fall close to each other.

In our MDS plot on pseudobulk counts (Fig. \@ref(fig:pb-mds)), we can observe that cell-populations (clusters) are separated quite well, while WT and stimulated samples (groups) are separated to a lesser (if any) extend.

```{r pb-mds, fig.wide = TRUE, fig.cap = "Pseudobulk-level MDS plot. Each point represent one cluster-sample instance;\npoints are colored by cluster ID and shaped by group ID."}
pbMDS(pb)
```

# DS analysis

```{r ds-analysis}
# specify design & contrast matrix
(design <- model.matrix(~ 0 + ei$group_id) %>% 
    set_rownames(ei$sample_id) %>% 
    set_colnames(levels(ei$group_id)))
(contrast <- makeContrasts("LPS-WT", levels = design))
system.time(res <- pbDS(sce, pb, design, contrast, verbose = FALSE))
```

# Save SCE & unfiltered results

```{r}
saveRDS(sce, file.path("output", "MAGL-SCE.rds"))
saveRDS(res, file.path("output", "MAGL-DS_res.rds"))
```