---
title: "Exploration of the LPS DEA"
author: 
- name: Pierre-Luc Germain
  affiliation: 
  - &IMLS Institute for Molecular Life Sciences, University of Zurich, Switzerland
  - &SIB Swiss Institute of Bioinformatics (SIB), University of Zurich, Switzerland
- name: Helena L. Crowell
  affiliation:
  - *IMLS
  - *SIB
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output:
  workflowr::wflow_html:
    toc: true
    code_folding: hide
---

```{r setup, include = FALSE}
library(BiocStyle)
knitr::opts_chunk$set(autodep = TRUE, cache = TRUE)
options(future.globals.maxSize = 2e3*1024^2)
```

### Load packages
```{r load-libs, collapse = FALSE, message = FALSE, warning = FALSE}
library(AnnotationDbi)
library(circlize)
library(ComplexHeatmap)
library(dplyr)
library(edgeR)
library(ggplot2)
library(M3C)
library(msigdbr)
library(muscat)
library(RColorBrewer)
library(SingleCellExperiment)
library(topGO)
library(org.Mm.eg.db)
library(purrr)
```

# Load data & results
```{r load-data, collapse = FALSE}
sce <- readRDS(file.path("output", "MAGL-SCE.rds"))
res <- readRDS(file.path("output", "MAGL-DS_res.rds"))

# store gene symbols & feature names
ss <- strsplit(rownames(sce), ".", fixed = TRUE)
rowData(sce)$symbol <- sapply(ss, .subset, 1)
rowData(sce)$feature <- sapply(ss, .subset, 2)
```

For easy accession, we store the character vectors of cluster and sample IDs, as well as the number of clusters and samples:

```{r}
nk <- length(kids <- set_names(levels(sce$cluster_id)))
ns <- length(sids <- set_names(levels(sce$sample_id)))
ng <- length(gids <- set_names(levels(sce$group_id)))
```

```{r aggregation}
# aggregation to pseudobulk sum-counts
pb <- aggregateData(sce, assay = "counts", fun = "sum")
```

```{r logFCs-to-SCE}
# construct SCE containing logFCs for ea. cluster
ref <- pb$group_id == "WT"
lfc <- lapply(assays(pb), function(u) {
    y <- DGEList(u)
    y <- calcNormFactors(y)
    lcpm <- log1p(cpm.DGEList(y))
    lfc <- lcpm - rowMeans(lcpm[, ref])
    SingleCellExperiment(lfc, colData = colData(pb))
}) %>% do.call(what = cbind)
lfc$cluster_id <- rep(assayNames(pb), each = ns)
```

```{r}
# round limits of x to nearest n
.get_brks <- function(x, n) {
    n <- 1 / n
    min <- floor(min(x) * n) / n
    max <- ceiling(max(x) * n) / n
    c(0, min, max)
}

# color palettes for heatmap, cluster, sample, group IDs, and # cells
hm_cols <- c("grey95", "blue", "red")
kcols <- set_names(CATALYST:::.cluster_cols[seq_len(nk)], kids)
scols <- set_names(CATALYST:::.cluster_cols[seq_len(ns) + nk], sids)
gcols <- set_names(c("royalblue", "orange"), gids)

n_cells <- c(t(metadata(pb)$n_cells))
max <- .get_brks(n_cells, 100)[3]
ncols <- colorRamp2(c(0, max), c("white", "black"))

# column annoation 
col_df <- data.frame(n_cells, 
    cluster_id = rep(kids, each = ns), 
    group_id = rep(pb$group_id, nk))
col_cols <- list(
    n_cells = ncols, cluster_id = kcols, 
    sample_id = scols, group_id = gcols)
col_anno <- HeatmapAnnotation(
    df = col_df, col = col_cols,
    show_annotation_name = FALSE)

# heatmap wrapper
.plot_hm <- function(sce, col, brks, col_anno = NULL, row_anno = NULL, 
    row_split = NULL, row_nms = FALSE, col_title = NULL, ...)
    Heatmap(assay(sce), col, name = "logFC", 
        column_title = col_title, cluster_columns = FALSE, 
        show_row_names = row_nms, show_column_names = FALSE,
        show_row_dend = FALSE, show_column_dend = FALSE,
        row_split = row_split, column_split = sce$cluster_id,
        top_annotation = col_anno, left_annotation = row_anno, 
        heatmap_legend_param = list(at = brks), 
        row_names_side = "left", ...)
```

# Comparison across cell types

**Q1: To what extent are the changes shared or cell-type specific?**

In order to be able to asses the overlap and magnitude of changes across cell types independent of variations in the significance, we plot a heatmap of the logFC, in each cell type, of the union of differentially-expressed genes:

```{r get-DE-genes}
# get DE genes for ea. cluster
tbl <- res$table[[1]]
tbl_fil <- lapply(tbl, filter, p_adj.loc < 0.01, abs(logFC) > 1)
de_gs_by_k <- map(tbl_fil, pull, "gene")
de_gs <- unique(unlist(de_gs_by_k))
```

## Consensus clustering of genes

```{r consensus-clustering, message = FALSE, results = "hide"}
cc <- M3C(t(assay(lfc[de_gs, ])), method = 2)
cc_ids <- cc$realdataresults[[3]]$assignments
cc_cols <- brewer.pal(length(unique(cc_ids)), "Set2")
names(cc_cols) <- unique(cc_ids)
```

```{r hm-DE-genes, fig.width = 14}
row_anno <- rowAnnotation(
    show_annotation_name = FALSE,
    df = data.frame(consensus_id = cc_ids),
    col = list(consensus_id = cc_cols))

brks <- .get_brks(assay(lfc[de_gs, ]), 0.5)
cols <- colorRamp2(brks, hm_cols)
.plot_hm(lfc[de_gs, ], cols, brks, col_anno, row_anno, cc_ids, row_title = NULL)
```

The pattern is very similar across cell subtypes of a given class, but there is also a striking similarity across all cell types, although neurons react much more mildly.

## GO over-representation analysis of clusters

```{r GO, collapse = FALSE, message = FALSE, warning = FALSE}
go <- lapply(split(names(cc_ids), cc_ids), function(gs) {
    ss <- strsplit(gs, ".", fixed = TRUE)
    fs <- unique(sapply(ss, .subset, 2))
    l <- as.numeric(rowData(sce)$feature %in% fs)
    names(l) <- rowData(sce)$feature
    go <- new("topGOdata", 
        ontology = "BP",
        allGenes = l,
        geneSel = function(u) u == 1,
        annot = annFUN.org,
        mapping = "org.Mm.eg.db",
        ID = "symbol")
    res <- runTest(go, algorithm = "classic", statistic = "fisher")
    GenTable(go, classicFisher = res, orderBy = "classicFisher", topNodes = 10)
})
```

# Conserved gene-cluster across cell types

```{r}
go[[3]]
```

The core response across all cell types is an upregulation of defense response genes, in particular, genes related to the response to cytokine and bacterium. We look specifically at the 'response to bacterium' genes:

```{r bacteria, fig.width = 14}
rbg <- mget("GO:0009617", org.Mm.egGO2ALLEGS)[[1]]
rbg <- unique(unlist(mget(rbg, org.Mm.egSYMBOL)))
rbg <- sapply(rbg, grep, rowData(sce)$feature)
rbg <- rownames(sce)[unlist(rbg)]
rbg <- intersect(rbg, de_gs)

brks <- .get_brks(assay(lfc[rbg, ]), 0.5)
cols <- colorRamp2(brks, hm_cols)
.plot_hm(lfc[rbg, ], cols, brks, col_anno)
```

Finally, a fairly specific GO term that comes up regularly in the camera analysis:

```{r chemokine, fig.width = 14, fig.height = 6}
set <- msigdbr(species = "Mus musculus")
set <- set[set$gs_name == "GO_CHEMOKINE_RECEPTOR_BINDING", ]
idx <- match(set$gene_symbol, rowData(sce)$feature, nomatch = 0)

brks <- .get_brks(assay(lfc[idx, ]), 0.5)
cols <- colorRamp2(brks, hm_cols)
.plot_hm(lfc[idx, ], cols, brks, col_anno, row_nms = TRUE, 
    col_title = "DE genes related to chemokine receptor binding")
```

# Cell-level linear decomposition

**Q2: Are the changes homogeneous across cells of a given type, or do they affect a subset of the cells?**

To understand whether alterations homogeneously affect all cells, or a subset of the cells, we plot, for each cell type,
the distribution of the cells' position on the first component of a PCA based on the DEGs.

Since neurons have few DEGs, we use the union of DEGs of the cell type (e.g. all excitatory neurons):

```{r linear-decomp, fig.width = 14}
cs_by_k <- split(colnames(sce), sce$cluster_id)
ld <- lapply(kids, function(k) {
    # subset SCE to inlcude
    # - genes that are DE in cluster k
    # - cells that have been assigned to cluster k
    gs <- de_gs_by_k[[k]]
    cs <- cs_by_k[[k]]
    es <- logcounts(sce[gs, cs])
    # decompose values as a linear combination of the group's logFCs
    ld <- t(es) %*% tbl_fil[[k]]$logFC
    # assure values are comparable across clusters
    ld[, 1] / max(ld)
}) %>% unlist %>% set_names(unlist(cs_by_k))

dim_red <- do.call(cbind, reducedDims(sce))
gg_df <- data.frame(colData(sce), ld = ld[colnames(sce)], dim_red)
```

## Violins by cluster-sample

```{r LD-by-ks, warning = FALSE, fig.width = 14}
ggplot(gg_df, aes(x = sample_id, y = ld, fill = group_id)) +
    facet_wrap("cluster_id", scales = "free_y", ncol = 7) +
    scale_fill_manual(values = gcols) + geom_violin() + 
    geom_boxplot(width = 0.1, outlier.size = 0.5, show.legend = FALSE) +
    labs(x = NULL, y = "effect coefficient") + theme_light() + 
    theme(aspect.ratio = 1, panel.grid.minor = element_blank(),
        axis.ticks=element_blank(), axis.text.x = element_blank())
```

For endothelial and most glial cells, virtually all cells appear to react similarly, with a clear separation between groups, 
although the larger spread of LPS-treated populations suggests that some cells are more affected than others. 
Instead, in neurons it appears that a fairly large proportion of the cells are hardly affected.

## UMAP cell embeddings

```{r UMAPs, fig.width = 12}
ggplot(gg_df, aes(x = UMAP_1, y = UMAP_2, col = ld)) + 
    geom_point(alpha = 0.4, size = 0.8) + 
    facet_wrap("sample_id", ncol = 4) +
    theme_void() + theme(aspect.ratio = 1)
```

As a proxy for citotoxicity, we check whether there is a relationship of the effect coefficient with the proportion of mitochondrial RNA:

## Effect coefs. vs. % mitochondrial

```{r LD-vs-pMT, warning = FALSE, fig.width = 14}
ggplot(gg_df, aes(x = ld, y = pct_counts_Mt, col = group_id)) + 
    geom_point() + facet_wrap("cluster_id", scales = "free", ncol = 7) +
    theme(aspect.ratio = 1)

gg_df[gg_df$group_id == "LPS", ] %>% 
    ggplot(aes(x = ld, y = pct_counts_Mt)) + 
    geom_point() + geom_density_2d() + 
    facet_wrap("cluster_id", scales = "free", ncol = 7) +
    xlab("effect coefficient") + ggtitle("Treated samples only") + 
    theme(aspect.ratio = 1, axis.text = element_blank(), axis.ticks = element_blank())
```

There isn't much correlation, but in excitatory neurons (6) there might be a rough association between the amount of mitochondrial RNA and the degree to which a cell is affected - in the others if anything it's the opposite.

## Effect coefs. vs. library sizes

The relationship with library size is worrying:

```{r LD-vs-counts, warning = FALSE, fig.width = 14}
gg_df[gg_df$group_id == "LPS", ] %>%
    ggplot(aes(x = ld, y = log10_total_counts)) + 
    geom_point() + geom_density_2d() + 
    facet_wrap("cluster_id", scales = "free", ncol = 7) + 
    xlab("effect coefficient") + ggtitle("Treated samples only") + 
    theme(aspect.ratio = 1, axis.text = element_blank(), axis.ticks = element_blank())
```

In neurons, when a cell has more reads it tends to be less affected by the treatment. A possible 
explanation for this would be that other cell types secrete a lot of RNA which is catched in droplets along with
the cells, and when a cell is small their presence is larger... So perhaps the global signature we see is in fact
an artefact?

***

# Expression of LPS response genes

The LPS receptor Cd14, various chemokine receptors and interferon regulatory factors, as well as the primiary transcriptional response genes Cxcl2 and Tnf, are all upregulated across cell types, suggesting that all cells sense threat. Tnf, which mediates the cytotoxic effect of LPS, is upregulated in a statistically-significant manner only in microglia (where it is also more abundant), but there is a trend in across all cell types, and the low detection rate prevents us from drawing clear conclusions. Various members of the interferon response and Nf-kb pathways are similarly upregulated across cell types. However, notable differences suggest distinct modalities of response orchestrated across cell types. While endothelial and glial cells upregulate several components of the Nf-$\kappa K$b (Rela, Relb) and Stat pathways (Stat1, Stat2, Stat3), only part of this response (chiefly Stat3 and to a lower extent Stat1) is visible in neurons. Only astrocytes (and possibly OPCs) show an upregulation of Tirap, involved in the the signal transduction of the Toll-like receptors, while only microglia upregulate Traf3 and Traf6. The LPS-induced TNF Factor (Litaf) and the early response gene Ptgs1 are upregulated only in endothelial cells. Endothelial cells and microglia upregulate Dusp1 and Dusp16, which are expected to dephosphorylate the MAPK1/10. Finally, while the immediate early stress response genes Fos, Jun and Junb are strongly upregulated in astrocytes, endothelial cells and OPCs, they show no such trend in neurons -- in fact, Fos suprisingly appears downregulated by neurons upon LPS exposure, along with another important immediate early gene, Egr1.

```{r fig.width = 12, fig.height = 6}
fs <- c("Tirap", "Jun", "Fos", "Cd83", "Traf6", "Traf3", "Sp3", "Litaf", "Egr1", 
    "Rela", "Ptgs2", "Junb", "Relb", "Cxcl2", "Tnf", "Cd14", "Cxcl10", "Irf7", 
    "Irf1", "Ccl5", "Tlr2", "Irf9", "Stat9", "Stat3", "Nbkbia", "Stat1", 
    "Stat2", "Fkbp5", "Gpx1", "Abca1", "Tnfrsf1a", "Dusp1", "Dusp16")
m <- sapply(fs, match, rowData(sce)$feature, nomatch = 0)

brks <- .get_brks(assay(lfc[m, ]), 0.5)
cols <- colorRamp2(brks, hm_cols)
.plot_hm(lfc[m, ], cols, brks, col_anno, row_nms = TRUE)
```

## Tnf

```{r Tnf, fig.width = 14}
g <- "ENSMUSG00000024401.Tnf"
es <- as.numeric(logcounts(sce[g, ]))
gg_df <- data.frame(colData(sce), expression = es)

ggplot(gg_df, aes(x = sample_id, y = expression, fill = group_id)) +
    facet_wrap("cluster_id", ncol = 7) + geom_violin() + 
    theme(aspect.ratio = 1, axis.text.x = element_blank(), axis.ticks.x = element_blank())
```

# Some notes

* There seems to be hardly any difference between the different clusters of inhibitory and excitatory neurons; therefore, after having shown this (e.g. through the first heatmap here), we could proceed with the rest using only the 7 main cell types.
* The molecular pathways from the receptors to TNF-alpha activity are protein signaling pathways, not transcriptional pathways (though downstream it gets transcriptional, and there's a transcriptional feed-forward loop at different steps). One should therefore remain cautious of drawing conclusions from RNA levels.
* With this caveat in mind, I was personally surprised to see much of the canonical response across all cell types, since I thought that most of it was happening in cells of the hematopoietic system and a few other cell types here and there. Of course, it could still be that this is non-cell-autonomous (secondary response mediated by other cell types), but as mentioned earlier it might also be contamination from floating RNA. It might be worth having a quick input of an immunologist on this (I'd know someone in case): perhaps the response in neurons is normal, but if it's unexpected then I guess one would need to investigate the possibility of contamination (going back to excluded droplets, and doing something like SoupX)...